// storybook/dist/backgrounds/preview.js
var O = Object.defineProperty;
var a = (e, t) => O(e, "name", { value: t, configurable: !0 });

// src/backgrounds/preview.ts
import { definePreview as j } from "storybook/preview-api";

// src/backgrounds/constants.ts
var U = "storybook/background", c = "backgrounds";
var Q = {
  UPDATE: `${U}/update`
};

// src/backgrounds/decorator.ts
import { useEffect as L } from "storybook/preview-api";

// src/backgrounds/defaults.ts
var R = {
  light: { name: "light", value: "#F8F8F8" },
  dark: { name: "dark", value: "#333" }
};

// src/backgrounds/utils.ts
var { document: n } = globalThis, D = /* @__PURE__ */ a(() => globalThis?.matchMedia ? !!globalThis.matchMedia("(prefers-reduced-motion: red\
uce)")?.matches : !1, "isReduceMotionEnabled"), k = /* @__PURE__ */ a((e) => {
  (Array.isArray(e) ? e : [e]).forEach(z);
}, "clearStyles"), z = /* @__PURE__ */ a((e) => {
  if (!n)
    return;
  let t = n.getElementById(e);
  t && t.parentElement && t.parentElement.removeChild(t);
}, "clearStyle"), P = /* @__PURE__ */ a((e, t) => {
  if (!n)
    return;
  let s = n.getElementById(e);
  if (s)
    s.innerHTML !== t && (s.innerHTML = t);
  else {
    let r = n.createElement("style");
    r.setAttribute("id", e), r.innerHTML = t, n.head.appendChild(r);
  }
}, "addGridStyle"), _ = /* @__PURE__ */ a((e, t, s) => {
  if (!n)
    return;
  let r = n.getElementById(e);
  if (r)
    r.innerHTML !== t && (r.innerHTML = t);
  else {
    let o = n.createElement("style");
    o.setAttribute("id", e), o.innerHTML = t;
    let d = `addon-backgrounds-grid${s ? `-docs-${s}` : ""}`, l = n.getElementById(d);
    l ? l.parentElement?.insertBefore(o, l) : n.head.appendChild(o);
  }
}, "addBackgroundStyle");

// src/backgrounds/decorator.ts
var N = {
  cellSize: 100,
  cellAmount: 10,
  opacity: 0.8
}, v = "addon-backgrounds", C = "addon-backgrounds-grid", Y = D() ? "" : "transition: background-color 0.3s;", w = /* @__PURE__ */ a((e, t) => {
  let { globals: s = {}, parameters: r = {}, viewMode: o, id: d } = t, {
    options: l = R,
    disable: F,
    grid: I = N
  } = r[c] || {}, p = s[c] || {}, S = typeof p == "string" ? p : p?.value, y = S ? l[S] : void 0, E = typeof y == "string" ? y : y?.value ||
  "transparent", A = typeof p == "string" ? !1 : p.grid || !1, h = !!y && !F, B = o === "docs" ? `#anchor--${d} .docs-story` : ".sb-show-mai\
n", M = o === "docs" ? `#anchor--${d} .docs-story` : ".sb-show-main", H = r.layout === void 0 || r.layout === "padded", T = o === "docs" ? 20 :
  H ? 16 : 0, { cellAmount: g, cellSize: i, opacity: u, offsetX: m = T, offsetY: f = T } = I, b = o === "docs" ? `${v}-docs-${d}` : `${v}-co\
lor`, G = o === "docs" ? d : null;
  L(() => {
    let $ = `
    ${B} {
      background: ${E} !important;
      ${Y}
      }`;
    if (!h) {
      k(b);
      return;
    }
    _(b, $, G);
  }, [B, b, G, h, E]);
  let x = o === "docs" ? `${C}-docs-${d}` : `${C}`;
  return L(() => {
    if (!A) {
      k(x);
      return;
    }
    let $ = [
      `${i * g}px ${i * g}px`,
      `${i * g}px ${i * g}px`,
      `${i}px ${i}px`,
      `${i}px ${i}px`
    ].join(", "), K = `
        ${M} {
          background-size: ${$} !important;
          background-position: ${m}px ${f}px, ${m}px ${f}px, ${m}px ${f}px, ${m}px ${f}px !important;
          background-blend-mode: difference !important;
          background-image: linear-gradient(rgba(130, 130, 130, ${u}) 1px, transparent 1px),
           linear-gradient(90deg, rgba(130, 130, 130, ${u}) 1px, transparent 1px),
           linear-gradient(rgba(130, 130, 130, ${u / 2}) 1px, transparent 1px),
           linear-gradient(90deg, rgba(130, 130, 130, ${u / 2}) 1px, transparent 1px) !important;
        }
      `;
    P(x, K);
  }, [g, i, M, x, A, m, f, u]), e();
}, "withBackgroundAndGrid");

// src/backgrounds/preview.ts
var V = globalThis.FEATURES?.backgrounds ? [w] : [], X = {
  [c]: {
    grid: {
      cellSize: 20,
      opacity: 0.5,
      cellAmount: 5
    },
    disable: !1
  }
}, q = {
  [c]: { value: void 0, grid: !1 }
}, pt = /* @__PURE__ */ a(() => j({
  decorators: V,
  parameters: X,
  initialGlobals: q
}), "default");
export {
  pt as default
};
