// storybook/dist/backgrounds/preview.cjs
"use strict";
var b = Object.defineProperty;
var z = Object.getOwnPropertyDescriptor;
var N = Object.getOwnPropertyNames;
var Y = Object.prototype.hasOwnProperty;
var i = (e, t) => b(e, "name", { value: t, configurable: !0 });
var j = (e, t) => {
  for (var n in t)
    b(e, n, { get: t[n], enumerable: !0 });
}, V = (e, t, n, o) => {
  if (t && typeof t == "object" || typeof t == "function")
    for (let r of N(t))
      !Y.call(e, r) && r !== n && b(e, r, { get: () => t[r], enumerable: !(o = z(t, r)) || o.enumerable });
  return e;
};
var X = (e) => V(b({}, "__esModule", { value: !0 }), e);

// src/backgrounds/preview.ts
var ot = {};
j(ot, {
  default: () => rt
});
module.exports = X(ot);
var I = require("storybook/preview-api");

// src/backgrounds/constants.ts
var q = "storybook/background", c = "backgrounds";
var at = {
  UPDATE: `${q}/update`
};

// src/backgrounds/decorator.ts
var E = require("storybook/preview-api");

// src/backgrounds/defaults.ts
var P = {
  light: { name: "light", value: "#F8F8F8" },
  dark: { name: "dark", value: "#333" }
};

// src/backgrounds/utils.ts
var { document: a } = globalThis, _ = /* @__PURE__ */ i(() => globalThis?.matchMedia ? !!globalThis.matchMedia("(prefers-reduced-motion: red\
uce)")?.matches : !1, "isReduceMotionEnabled"), S = /* @__PURE__ */ i((e) => {
  (Array.isArray(e) ? e : [e]).forEach(J);
}, "clearStyles"), J = /* @__PURE__ */ i((e) => {
  if (!a)
    return;
  let t = a.getElementById(e);
  t && t.parentElement && t.parentElement.removeChild(t);
}, "clearStyle"), L = /* @__PURE__ */ i((e, t) => {
  if (!a)
    return;
  let n = a.getElementById(e);
  if (n)
    n.innerHTML !== t && (n.innerHTML = t);
  else {
    let o = a.createElement("style");
    o.setAttribute("id", e), o.innerHTML = t, a.head.appendChild(o);
  }
}, "addGridStyle"), v = /* @__PURE__ */ i((e, t, n) => {
  if (!a)
    return;
  let o = a.getElementById(e);
  if (o)
    o.innerHTML !== t && (o.innerHTML = t);
  else {
    let r = a.createElement("style");
    r.setAttribute("id", e), r.innerHTML = t;
    let d = `addon-backgrounds-grid${n ? `-docs-${n}` : ""}`, l = a.getElementById(d);
    l ? l.parentElement?.insertBefore(r, l) : a.head.appendChild(r);
  }
}, "addBackgroundStyle");

// src/backgrounds/decorator.ts
var Q = {
  cellSize: 100,
  cellAmount: 10,
  opacity: 0.8
}, C = "addon-backgrounds", w = "addon-backgrounds-grid", W = _() ? "" : "transition: background-color 0.3s;", F = /* @__PURE__ */ i((e, t) => {
  let { globals: n = {}, parameters: o = {}, viewMode: r, id: d } = t, {
    options: l = P,
    disable: H,
    grid: K = Q
  } = o[c] || {}, p = n[c] || {}, A = typeof p == "string" ? p : p?.value, y = A ? l[A] : void 0, h = typeof y == "string" ? y : y?.value ||
  "transparent", B = typeof p == "string" ? !1 : p.grid || !1, M = !!y && !H, T = r === "docs" ? `#anchor--${d} .docs-story` : ".sb-show-mai\
n", G = r === "docs" ? `#anchor--${d} .docs-story` : ".sb-show-main", O = o.layout === void 0 || o.layout === "padded", R = r === "docs" ? 20 :
  O ? 16 : 0, { cellAmount: g, cellSize: s, opacity: u, offsetX: m = R, offsetY: f = R } = K, x = r === "docs" ? `${C}-docs-${d}` : `${C}-co\
lor`, D = r === "docs" ? d : null;
  (0, E.useEffect)(() => {
    let k = `
    ${T} {
      background: ${h} !important;
      ${W}
      }`;
    if (!M) {
      S(x);
      return;
    }
    v(x, k, D);
  }, [T, x, D, M, h]);
  let $ = r === "docs" ? `${w}-docs-${d}` : `${w}`;
  return (0, E.useEffect)(() => {
    if (!B) {
      S($);
      return;
    }
    let k = [
      `${s * g}px ${s * g}px`,
      `${s * g}px ${s * g}px`,
      `${s}px ${s}px`,
      `${s}px ${s}px`
    ].join(", "), U = `
        ${G} {
          background-size: ${k} !important;
          background-position: ${m}px ${f}px, ${m}px ${f}px, ${m}px ${f}px, ${m}px ${f}px !important;
          background-blend-mode: difference !important;
          background-image: linear-gradient(rgba(130, 130, 130, ${u}) 1px, transparent 1px),
           linear-gradient(90deg, rgba(130, 130, 130, ${u}) 1px, transparent 1px),
           linear-gradient(rgba(130, 130, 130, ${u / 2}) 1px, transparent 1px),
           linear-gradient(90deg, rgba(130, 130, 130, ${u / 2}) 1px, transparent 1px) !important;
        }
      `;
    L($, U);
  }, [g, s, G, $, B, m, f, u]), e();
}, "withBackgroundAndGrid");

// src/backgrounds/preview.ts
var Z = globalThis.FEATURES?.backgrounds ? [F] : [], tt = {
  [c]: {
    grid: {
      cellSize: 20,
      opacity: 0.5,
      cellAmount: 5
    },
    disable: !1
  }
}, et = {
  [c]: { value: void 0, grid: !1 }
}, rt = /* @__PURE__ */ i(() => (0, I.definePreview)({
  decorators: Z,
  parameters: tt,
  initialGlobals: et
}), "default");
