// storybook/dist/instrumenter/index.cjs
"use strict";
var Pe = Object.defineProperty;
var gr = Object.getOwnPropertyDescriptor;
var hr = Object.getOwnPropertyNames;
var yr = Object.prototype.hasOwnProperty;
var c = (e, t) => Pe(e, "name", { value: t, configurable: !0 });
var dr = (e, t) => {
  for (var n in t)
    Pe(e, n, { get: t[n], enumerable: !0 });
}, br = (e, t, n, r) => {
  if (t && typeof t == "object" || typeof t == "function")
    for (let o of hr(t))
      !yr.call(e, o) && o !== n && Pe(e, o, { get: () => t[o], enumerable: !(r = gr(t, o)) || r.enumerable });
  return e;
};
var Sr = (e) => br(Pe({}, "__esModule", { value: !0 }), e);

// src/instrumenter/index.ts
var cs = {};
dr(cs, {
  CallStates: () => cr,
  EVENTS: () => ee,
  instrument: () => mr
});
module.exports = Sr(cs);

// src/instrumenter/instrumenter.ts
var fr = require("storybook/internal/client-logger"), ye = require("storybook/internal/core-events"), W = require("@storybook/global");

// ../node_modules/tinyrainbow/dist/chunk-BVHSVHOK.js
var Er = {
  reset: [0, 0],
  bold: [1, 22, "\x1B[22m\x1B[1m"],
  dim: [2, 22, "\x1B[22m\x1B[2m"],
  italic: [3, 23],
  underline: [4, 24],
  inverse: [7, 27],
  hidden: [8, 28],
  strikethrough: [9, 29],
  black: [30, 39],
  red: [31, 39],
  green: [32, 39],
  yellow: [33, 39],
  blue: [34, 39],
  magenta: [35, 39],
  cyan: [36, 39],
  white: [37, 39],
  gray: [90, 39],
  bgBlack: [40, 49],
  bgRed: [41, 49],
  bgGreen: [42, 49],
  bgYellow: [43, 49],
  bgBlue: [44, 49],
  bgMagenta: [45, 49],
  bgCyan: [46, 49],
  bgWhite: [47, 49],
  blackBright: [90, 39],
  redBright: [91, 39],
  greenBright: [92, 39],
  yellowBright: [93, 39],
  blueBright: [94, 39],
  magentaBright: [95, 39],
  cyanBright: [96, 39],
  whiteBright: [97, 39],
  bgBlackBright: [100, 49],
  bgRedBright: [101, 49],
  bgGreenBright: [102, 49],
  bgYellowBright: [103, 49],
  bgBlueBright: [104, 49],
  bgMagentaBright: [105, 49],
  bgCyanBright: [106, 49],
  bgWhiteBright: [107, 49]
}, _r = Object.entries(Er);
function Qe(e) {
  return String(e);
}
c(Qe, "a");
Qe.open = "";
Qe.close = "";
function xt(e = !1) {
  let t = typeof process < "u" ? process : void 0, n = t?.env || {}, r = t?.argv || [];
  return !("NO_COLOR" in n || r.includes("--no-color")) && ("FORCE_COLOR" in n || r.includes("--color") || t?.platform === "win32" || e && n.
  TERM !== "dumb" || "CI" in n) || typeof window < "u" && !!window.chrome;
}
c(xt, "C");
function Dt(e = !1) {
  let t = xt(e), n = /* @__PURE__ */ c((s, a, l, m) => {
    let p = "", f = 0;
    do
      p += s.substring(f, m) + l, f = m + a.length, m = s.indexOf(a, f);
    while (~m);
    return p + s.substring(f);
  }, "i"), r = /* @__PURE__ */ c((s, a, l = s) => {
    let m = /* @__PURE__ */ c((p) => {
      let f = String(p), b = f.indexOf(a, s.length);
      return ~b ? s + n(f, a, l, b) + a : s + f + a;
    }, "o");
    return m.open = s, m.close = a, m;
  }, "g"), o = {
    isColorSupported: t
  }, i = /* @__PURE__ */ c((s) => `\x1B[${s}m`, "d");
  for (let [s, a] of _r)
    o[s] = t ? r(
      i(a[0]),
      i(a[1]),
      a[2]
    ) : Qe;
  return o;
}
c(Dt, "p");

// ../node_modules/tinyrainbow/dist/node.js
var Ft = require("tty");
var Tr = process.env.FORCE_TTY !== void 0 || (0, Ft.isatty)(1);
var ne = Dt(Tr);

// ../node_modules/@vitest/pretty-format/dist/index.js
function Zt(e, t) {
  return t.forEach(function(n) {
    n && typeof n != "string" && !Array.isArray(n) && Object.keys(n).forEach(function(r) {
      if (r !== "default" && !(r in e)) {
        var o = Object.getOwnPropertyDescriptor(n, r);
        Object.defineProperty(e, r, o.get ? o : {
          enumerable: !0,
          get: /* @__PURE__ */ c(function() {
            return n[r];
          }, "get")
        });
      }
    });
  }), Object.freeze(e);
}
c(Zt, "_mergeNamespaces");
function Cr(e, t) {
  let n = Object.keys(e), r = t === null ? n : n.sort(t);
  if (Object.getOwnPropertySymbols)
    for (let o of Object.getOwnPropertySymbols(e))
      Object.getOwnPropertyDescriptor(e, o).enumerable && r.push(o);
  return r;
}
c(Cr, "getKeysOfEnumerableProperties");
function Oe(e, t, n, r, o, i, s = ": ") {
  let a = "", l = 0, m = e.next();
  if (!m.done) {
    a += t.spacingOuter;
    let p = n + t.indent;
    for (; !m.done; ) {
      if (a += p, l++ === t.maxWidth) {
        a += "\u2026";
        break;
      }
      let f = i(m.value[0], t, p, r, o), b = i(m.value[1], t, p, r, o);
      a += f + s + b, m = e.next(), m.done ? t.min || (a += ",") : a += `,${t.spacingInner}`;
    }
    a += t.spacingOuter + n;
  }
  return a;
}
c(Oe, "printIteratorEntries");
function nt(e, t, n, r, o, i) {
  let s = "", a = 0, l = e.next();
  if (!l.done) {
    s += t.spacingOuter;
    let m = n + t.indent;
    for (; !l.done; ) {
      if (s += m, a++ === t.maxWidth) {
        s += "\u2026";
        break;
      }
      s += i(l.value, t, m, r, o), l = e.next(), l.done ? t.min || (s += ",") : s += `,${t.spacingInner}`;
    }
    s += t.spacingOuter + n;
  }
  return s;
}
c(nt, "printIteratorValues");
function De(e, t, n, r, o, i) {
  let s = "";
  e = e instanceof ArrayBuffer ? new DataView(e) : e;
  let a = /* @__PURE__ */ c((m) => m instanceof DataView, "isDataView"), l = a(e) ? e.byteLength : e.length;
  if (l > 0) {
    s += t.spacingOuter;
    let m = n + t.indent;
    for (let p = 0; p < l; p++) {
      if (s += m, p === t.maxWidth) {
        s += "\u2026";
        break;
      }
      (a(e) || p in e) && (s += i(a(e) ? e.getInt8(p) : e[p], t, m, r, o)), p < l - 1 ? s += `,${t.spacingInner}` : t.min || (s += ",");
    }
    s += t.spacingOuter + n;
  }
  return s;
}
c(De, "printListItems");
function rt(e, t, n, r, o, i) {
  let s = "", a = Cr(e, t.compareKeys);
  if (a.length > 0) {
    s += t.spacingOuter;
    let l = n + t.indent;
    for (let m = 0; m < a.length; m++) {
      let p = a[m], f = i(p, t, l, r, o), b = i(e[p], t, l, r, o);
      s += `${l + f}: ${b}`, m < a.length - 1 ? s += `,${t.spacingInner}` : t.min || (s += ",");
    }
    s += t.spacingOuter + n;
  }
  return s;
}
c(rt, "printObjectProperties");
var Or = typeof Symbol == "function" && Symbol.for ? Symbol.for("jest.asymmetricMatcher") : 1267621, Ie = " ", $r = /* @__PURE__ */ c((e, t, n, r, o, i) => {
  let s = e.toString();
  if (s === "ArrayContaining" || s === "ArrayNotContaining")
    return ++r > t.maxDepth ? `[${s}]` : `${s + Ie}[${De(e.sample, t, n, r, o, i)}]`;
  if (s === "ObjectContaining" || s === "ObjectNotContaining")
    return ++r > t.maxDepth ? `[${s}]` : `${s + Ie}{${rt(e.sample, t, n, r, o, i)}}`;
  if (s === "StringMatching" || s === "StringNotMatching" || s === "StringContaining" || s === "StringNotContaining")
    return s + Ie + i(e.sample, t, n, r, o);
  if (typeof e.toAsymmetricMatcher != "function")
    throw new TypeError(`Asymmetric matcher ${e.constructor.name} does not implement toAsymmetricMatcher()`);
  return e.toAsymmetricMatcher();
}, "serialize$5"), wr = /* @__PURE__ */ c((e) => e && e.$$typeof === Or, "test$5"), Ar = {
  serialize: $r,
  test: wr
}, Rr = " ", vt = /* @__PURE__ */ new Set(["DOMStringMap", "NamedNodeMap"]), Nr = /^(?:HTML\w*Collection|NodeList)$/;
function Pr(e) {
  return vt.has(e) || Nr.test(e);
}
c(Pr, "testName");
var Ir = /* @__PURE__ */ c((e) => e && e.constructor && !!e.constructor.name && Pr(e.constructor.name), "test$4");
function Mr(e) {
  return e.constructor.name === "NamedNodeMap";
}
c(Mr, "isNamedNodeMap");
var Lr = /* @__PURE__ */ c((e, t, n, r, o, i) => {
  let s = e.constructor.name;
  return ++r > t.maxDepth ? `[${s}]` : (t.min ? "" : s + Rr) + (vt.has(s) ? `{${rt(Mr(e) ? [...e].reduce((a, l) => (a[l.name] = l.value, a),
  {}) : { ...e }, t, n, r, o, i)}}` : `[${De([...e], t, n, r, o, i)}]`);
}, "serialize$4"), xr = {
  serialize: Lr,
  test: Ir
};
function Qt(e) {
  return e.replaceAll("<", "&lt;").replaceAll(">", "&gt;");
}
c(Qt, "escapeHTML");
function ot(e, t, n, r, o, i, s) {
  let a = r + n.indent, l = n.colors;
  return e.map((m) => {
    let p = t[m], f = s(p, n, a, o, i);
    return typeof p != "string" && (f.includes(`
`) && (f = n.spacingOuter + a + f + n.spacingOuter + r), f = `{${f}}`), `${n.spacingInner + r + l.prop.open + m + l.prop.close}=${l.value.open}${f}${l.
    value.close}`;
  }).join("");
}
c(ot, "printProps");
function it(e, t, n, r, o, i) {
  return e.map((s) => t.spacingOuter + n + (typeof s == "string" ? en(s, t) : i(s, t, n, r, o))).join("");
}
c(it, "printChildren");
function en(e, t) {
  let n = t.colors.content;
  return n.open + Qt(e) + n.close;
}
c(en, "printText");
function Dr(e, t) {
  let n = t.colors.comment;
  return `${n.open}<!--${Qt(e)}-->${n.close}`;
}
c(Dr, "printComment");
function st(e, t, n, r, o) {
  let i = r.colors.tag;
  return `${i.open}<${e}${t && i.close + t + r.spacingOuter + o + i.open}${n ? `>${i.close}${n}${r.spacingOuter}${o}${i.open}</${e}` : `${t &&
  !r.min ? "" : " "}/`}>${i.close}`;
}
c(st, "printElement");
function ct(e, t) {
  let n = t.colors.tag;
  return `${n.open}<${e}${n.close} \u2026${n.open} />${n.close}`;
}
c(ct, "printElementAsLeaf");
var Fr = 1, tn = 3, nn = 8, rn = 11, jr = /^(?:(?:HTML|SVG)\w*)?Element$/;
function Br(e) {
  try {
    return typeof e.hasAttribute == "function" && e.hasAttribute("is");
  } catch {
    return !1;
  }
}
c(Br, "testHasAttribute");
function kr(e) {
  let t = e.constructor.name, { nodeType: n, tagName: r } = e, o = typeof r == "string" && r.includes("-") || Br(e);
  return n === Fr && (jr.test(t) || o) || n === tn && t === "Text" || n === nn && t === "Comment" || n === rn && t === "DocumentFragment";
}
c(kr, "testNode");
var zr = /* @__PURE__ */ c((e) => {
  var t;
  return (e == null || (t = e.constructor) === null || t === void 0 ? void 0 : t.name) && kr(e);
}, "test$3");
function Yr(e) {
  return e.nodeType === tn;
}
c(Yr, "nodeIsText");
function Ur(e) {
  return e.nodeType === nn;
}
c(Ur, "nodeIsComment");
function et(e) {
  return e.nodeType === rn;
}
c(et, "nodeIsFragment");
var Wr = /* @__PURE__ */ c((e, t, n, r, o, i) => {
  if (Yr(e))
    return en(e.data, t);
  if (Ur(e))
    return Dr(e.data, t);
  let s = et(e) ? "DocumentFragment" : e.tagName.toLowerCase();
  return ++r > t.maxDepth ? ct(s, t) : st(s, ot(et(e) ? [] : Array.from(e.attributes, (a) => a.name).sort(), et(e) ? {} : [...e.attributes].
  reduce((a, l) => (a[l.name] = l.value, a), {}), t, n + t.indent, r, o, i), it(Array.prototype.slice.call(e.childNodes || e.children), t, n +
  t.indent, r, o, i), t, n);
}, "serialize$3"), Vr = {
  serialize: Wr,
  test: zr
}, qr = "@@__IMMUTABLE_ITERABLE__@@", Kr = "@@__IMMUTABLE_LIST__@@", Gr = "@@__IMMUTABLE_KEYED__@@", Hr = "@@__IMMUTABLE_MAP__@@", jt = "@@_\
_IMMUTABLE_ORDERED__@@", Jr = "@@__IMMUTABLE_RECORD__@@", Xr = "@@__IMMUTABLE_SEQ__@@", Zr = "@@__IMMUTABLE_SET__@@", vr = "@@__IMMUTABLE_ST\
ACK__@@", be = /* @__PURE__ */ c((e) => `Immutable.${e}`, "getImmutableName"), je = /* @__PURE__ */ c((e) => `[${e}]`, "printAsLeaf"), Ce = "\
 ", Bt = "\u2026";
function Qr(e, t, n, r, o, i, s) {
  return ++r > t.maxDepth ? je(be(s)) : `${be(s) + Ce}{${Oe(e.entries(), t, n, r, o, i)}}`;
}
c(Qr, "printImmutableEntries");
function eo(e) {
  let t = 0;
  return { next() {
    if (t < e._keys.length) {
      let n = e._keys[t++];
      return {
        done: !1,
        value: [n, e.get(n)]
      };
    }
    return {
      done: !0,
      value: void 0
    };
  } };
}
c(eo, "getRecordEntries");
function to(e, t, n, r, o, i) {
  let s = be(e._name || "Record");
  return ++r > t.maxDepth ? je(s) : `${s + Ce}{${Oe(eo(e), t, n, r, o, i)}}`;
}
c(to, "printImmutableRecord");
function no(e, t, n, r, o, i) {
  let s = be("Seq");
  return ++r > t.maxDepth ? je(s) : e[Gr] ? `${s + Ce}{${e._iter || e._object ? Oe(e.entries(), t, n, r, o, i) : Bt}}` : `${s + Ce}[${e._iter ||
  e._array || e._collection || e._iterable ? nt(e.values(), t, n, r, o, i) : Bt}]`;
}
c(no, "printImmutableSeq");
function tt(e, t, n, r, o, i, s) {
  return ++r > t.maxDepth ? je(be(s)) : `${be(s) + Ce}[${nt(e.values(), t, n, r, o, i)}]`;
}
c(tt, "printImmutableValues");
var ro = /* @__PURE__ */ c((e, t, n, r, o, i) => e[Hr] ? Qr(e, t, n, r, o, i, e[jt] ? "OrderedMap" : "Map") : e[Kr] ? tt(e, t, n, r, o, i, "\
List") : e[Zr] ? tt(e, t, n, r, o, i, e[jt] ? "OrderedSet" : "Set") : e[vr] ? tt(e, t, n, r, o, i, "Stack") : e[Xr] ? no(e, t, n, r, o, i) :
to(e, t, n, r, o, i), "serialize$2"), oo = /* @__PURE__ */ c((e) => e && (e[qr] === !0 || e[Jr] === !0), "test$2"), io = {
  serialize: ro,
  test: oo
};
function on(e) {
  return e && e.__esModule && Object.prototype.hasOwnProperty.call(e, "default") ? e.default : e;
}
c(on, "getDefaultExportFromCjs");
var Me = { exports: {} }, N = {};
var kt;
function so() {
  if (kt) return N;
  kt = 1;
  var e = Symbol.for("react.transitional.element"), t = Symbol.for("react.portal"), n = Symbol.for("react.fragment"), r = Symbol.for("react.\
strict_mode"), o = Symbol.for("react.profiler"), i = Symbol.for("react.consumer"), s = Symbol.for("react.context"), a = Symbol.for("react.fo\
rward_ref"), l = Symbol.for("react.suspense"), m = Symbol.for("react.suspense_list"), p = Symbol.for("react.memo"), f = Symbol.for("react.la\
zy"), b = Symbol.for("react.view_transition"), y = Symbol.for("react.client.reference");
  function g(u) {
    if (typeof u == "object" && u !== null) {
      var h = u.$$typeof;
      switch (h) {
        case e:
          switch (u = u.type, u) {
            case n:
            case o:
            case r:
            case l:
            case m:
            case b:
              return u;
            default:
              switch (u = u && u.$$typeof, u) {
                case s:
                case a:
                case f:
                case p:
                  return u;
                case i:
                  return u;
                default:
                  return h;
              }
          }
        case t:
          return h;
      }
    }
  }
  return c(g, "typeOf"), N.ContextConsumer = i, N.ContextProvider = s, N.Element = e, N.ForwardRef = a, N.Fragment = n, N.Lazy = f, N.Memo =
  p, N.Portal = t, N.Profiler = o, N.StrictMode = r, N.Suspense = l, N.SuspenseList = m, N.isContextConsumer = function(u) {
    return g(u) === i;
  }, N.isContextProvider = function(u) {
    return g(u) === s;
  }, N.isElement = function(u) {
    return typeof u == "object" && u !== null && u.$$typeof === e;
  }, N.isForwardRef = function(u) {
    return g(u) === a;
  }, N.isFragment = function(u) {
    return g(u) === n;
  }, N.isLazy = function(u) {
    return g(u) === f;
  }, N.isMemo = function(u) {
    return g(u) === p;
  }, N.isPortal = function(u) {
    return g(u) === t;
  }, N.isProfiler = function(u) {
    return g(u) === o;
  }, N.isStrictMode = function(u) {
    return g(u) === r;
  }, N.isSuspense = function(u) {
    return g(u) === l;
  }, N.isSuspenseList = function(u) {
    return g(u) === m;
  }, N.isValidElementType = function(u) {
    return typeof u == "string" || typeof u == "function" || u === n || u === o || u === r || u === l || u === m || typeof u == "object" && u !==
    null && (u.$$typeof === f || u.$$typeof === p || u.$$typeof === s || u.$$typeof === i || u.$$typeof === a || u.$$typeof === y || u.getModuleId !==
    void 0);
  }, N.typeOf = g, N;
}
c(so, "requireReactIs_production");
var P = {};
var zt;
function co() {
  return zt || (zt = 1, process.env.NODE_ENV !== "production" && function() {
    function e(u) {
      if (typeof u == "object" && u !== null) {
        var h = u.$$typeof;
        switch (h) {
          case t:
            switch (u = u.type, u) {
              case r:
              case i:
              case o:
              case m:
              case p:
              case y:
                return u;
              default:
                switch (u = u && u.$$typeof, u) {
                  case a:
                  case l:
                  case b:
                  case f:
                    return u;
                  case s:
                    return u;
                  default:
                    return h;
                }
            }
          case n:
            return h;
        }
      }
    }
    c(e, "typeOf");
    var t = Symbol.for("react.transitional.element"), n = Symbol.for("react.portal"), r = Symbol.for("react.fragment"), o = Symbol.for("reac\
t.strict_mode"), i = Symbol.for("react.profiler"), s = Symbol.for("react.consumer"), a = Symbol.for("react.context"), l = Symbol.for("react.\
forward_ref"), m = Symbol.for("react.suspense"), p = Symbol.for("react.suspense_list"), f = Symbol.for("react.memo"), b = Symbol.for("react.\
lazy"), y = Symbol.for("react.view_transition"), g = Symbol.for("react.client.reference");
    P.ContextConsumer = s, P.ContextProvider = a, P.Element = t, P.ForwardRef = l, P.Fragment = r, P.Lazy = b, P.Memo = f, P.Portal = n, P.Profiler =
    i, P.StrictMode = o, P.Suspense = m, P.SuspenseList = p, P.isContextConsumer = function(u) {
      return e(u) === s;
    }, P.isContextProvider = function(u) {
      return e(u) === a;
    }, P.isElement = function(u) {
      return typeof u == "object" && u !== null && u.$$typeof === t;
    }, P.isForwardRef = function(u) {
      return e(u) === l;
    }, P.isFragment = function(u) {
      return e(u) === r;
    }, P.isLazy = function(u) {
      return e(u) === b;
    }, P.isMemo = function(u) {
      return e(u) === f;
    }, P.isPortal = function(u) {
      return e(u) === n;
    }, P.isProfiler = function(u) {
      return e(u) === i;
    }, P.isStrictMode = function(u) {
      return e(u) === o;
    }, P.isSuspense = function(u) {
      return e(u) === m;
    }, P.isSuspenseList = function(u) {
      return e(u) === p;
    }, P.isValidElementType = function(u) {
      return typeof u == "string" || typeof u == "function" || u === r || u === i || u === o || u === m || u === p || typeof u == "object" &&
      u !== null && (u.$$typeof === b || u.$$typeof === f || u.$$typeof === a || u.$$typeof === s || u.$$typeof === l || u.$$typeof === g ||
      u.getModuleId !== void 0);
    }, P.typeOf = e;
  }()), P;
}
c(co, "requireReactIs_development$1");
var Yt;
function uo() {
  return Yt || (Yt = 1, process.env.NODE_ENV === "production" ? Me.exports = so() : Me.exports = co()), Me.exports;
}
c(uo, "requireReactIs$1");
var sn = uo(), lo = /* @__PURE__ */ on(sn), ao = /* @__PURE__ */ Zt({
  __proto__: null,
  default: lo
}, [sn]), Le = { exports: {} }, w = {};
var Ut;
function fo() {
  if (Ut) return w;
  Ut = 1;
  var e = Symbol.for("react.element"), t = Symbol.for("react.portal"), n = Symbol.for("react.fragment"), r = Symbol.for("react.strict_mode"),
  o = Symbol.for("react.profiler"), i = Symbol.for("react.provider"), s = Symbol.for("react.context"), a = Symbol.for("react.server_context"),
  l = Symbol.for("react.forward_ref"), m = Symbol.for("react.suspense"), p = Symbol.for("react.suspense_list"), f = Symbol.for("react.memo"),
  b = Symbol.for("react.lazy"), y = Symbol.for("react.offscreen"), g;
  g = Symbol.for("react.module.reference");
  function u(h) {
    if (typeof h == "object" && h !== null) {
      var S = h.$$typeof;
      switch (S) {
        case e:
          switch (h = h.type, h) {
            case n:
            case o:
            case r:
            case m:
            case p:
              return h;
            default:
              switch (h = h && h.$$typeof, h) {
                case a:
                case s:
                case l:
                case b:
                case f:
                case i:
                  return h;
                default:
                  return S;
              }
          }
        case t:
          return S;
      }
    }
  }
  return c(u, "v"), w.ContextConsumer = s, w.ContextProvider = i, w.Element = e, w.ForwardRef = l, w.Fragment = n, w.Lazy = b, w.Memo = f, w.
  Portal = t, w.Profiler = o, w.StrictMode = r, w.Suspense = m, w.SuspenseList = p, w.isAsyncMode = function() {
    return !1;
  }, w.isConcurrentMode = function() {
    return !1;
  }, w.isContextConsumer = function(h) {
    return u(h) === s;
  }, w.isContextProvider = function(h) {
    return u(h) === i;
  }, w.isElement = function(h) {
    return typeof h == "object" && h !== null && h.$$typeof === e;
  }, w.isForwardRef = function(h) {
    return u(h) === l;
  }, w.isFragment = function(h) {
    return u(h) === n;
  }, w.isLazy = function(h) {
    return u(h) === b;
  }, w.isMemo = function(h) {
    return u(h) === f;
  }, w.isPortal = function(h) {
    return u(h) === t;
  }, w.isProfiler = function(h) {
    return u(h) === o;
  }, w.isStrictMode = function(h) {
    return u(h) === r;
  }, w.isSuspense = function(h) {
    return u(h) === m;
  }, w.isSuspenseList = function(h) {
    return u(h) === p;
  }, w.isValidElementType = function(h) {
    return typeof h == "string" || typeof h == "function" || h === n || h === o || h === r || h === m || h === p || h === y || typeof h == "\
object" && h !== null && (h.$$typeof === b || h.$$typeof === f || h.$$typeof === i || h.$$typeof === s || h.$$typeof === l || h.$$typeof ===
    g || h.getModuleId !== void 0);
  }, w.typeOf = u, w;
}
c(fo, "requireReactIs_production_min");
var A = {};
var Wt;
function mo() {
  return Wt || (Wt = 1, process.env.NODE_ENV !== "production" && function() {
    var e = Symbol.for("react.element"), t = Symbol.for("react.portal"), n = Symbol.for("react.fragment"), r = Symbol.for("react.strict_mode"),
    o = Symbol.for("react.profiler"), i = Symbol.for("react.provider"), s = Symbol.for("react.context"), a = Symbol.for("react.server_contex\
t"), l = Symbol.for("react.forward_ref"), m = Symbol.for("react.suspense"), p = Symbol.for("react.suspense_list"), f = Symbol.for("react.mem\
o"), b = Symbol.for("react.lazy"), y = Symbol.for("react.offscreen"), g = !1, u = !1, h = !1, S = !1, _ = !1, O;
    O = Symbol.for("react.module.reference");
    function d(C) {
      return !!(typeof C == "string" || typeof C == "function" || C === n || C === o || _ || C === r || C === m || C === p || S || C === y ||
      g || u || h || typeof C == "object" && C !== null && (C.$$typeof === b || C.$$typeof === f || C.$$typeof === i || C.$$typeof === s || C.
      $$typeof === l || // This needs to include all possible module reference object
      // types supported by any Flight configuration anywhere since
      // we don't know which Flight build this will end up being used
      // with.
      C.$$typeof === O || C.getModuleId !== void 0));
    }
    c(d, "isValidElementType");
    function E(C) {
      if (typeof C == "object" && C !== null) {
        var ve = C.$$typeof;
        switch (ve) {
          case e:
            var Ne = C.type;
            switch (Ne) {
              case n:
              case o:
              case r:
              case m:
              case p:
                return Ne;
              default:
                var Lt = Ne && Ne.$$typeof;
                switch (Lt) {
                  case a:
                  case s:
                  case l:
                  case b:
                  case f:
                  case i:
                    return Lt;
                  default:
                    return ve;
                }
            }
          case t:
            return ve;
        }
      }
    }
    c(E, "typeOf");
    var $ = s, T = i, R = e, J = l, te = n, L = b, z = f, X = t, V = o, M = r, D = m, F = p, Z = !1, B = !1;
    function K(C) {
      return Z || (Z = !0, console.warn("The ReactIs.isAsyncMode() alias has been deprecated, and will be removed in React 18+.")), !1;
    }
    c(K, "isAsyncMode");
    function ie(C) {
      return B || (B = !0, console.warn("The ReactIs.isConcurrentMode() alias has been deprecated, and will be removed in React 18+.")), !1;
    }
    c(ie, "isConcurrentMode");
    function G(C) {
      return E(C) === s;
    }
    c(G, "isContextConsumer");
    function H(C) {
      return E(C) === i;
    }
    c(H, "isContextProvider");
    function ce(C) {
      return typeof C == "object" && C !== null && C.$$typeof === e;
    }
    c(ce, "isElement");
    function v(C) {
      return E(C) === l;
    }
    c(v, "isForwardRef");
    function q(C) {
      return E(C) === n;
    }
    c(q, "isFragment");
    function se(C) {
      return E(C) === b;
    }
    c(se, "isLazy");
    function de(C) {
      return E(C) === f;
    }
    c(de, "isMemo");
    function ae(C) {
      return E(C) === t;
    }
    c(ae, "isPortal");
    function Te(C) {
      return E(C) === o;
    }
    c(Te, "isProfiler");
    function Ae(C) {
      return E(C) === r;
    }
    c(Ae, "isStrictMode");
    function Re(C) {
      return E(C) === m;
    }
    c(Re, "isSuspense");
    function pr(C) {
      return E(C) === p;
    }
    c(pr, "isSuspenseList"), A.ContextConsumer = $, A.ContextProvider = T, A.Element = R, A.ForwardRef = J, A.Fragment = te, A.Lazy = L, A.Memo =
    z, A.Portal = X, A.Profiler = V, A.StrictMode = M, A.Suspense = D, A.SuspenseList = F, A.isAsyncMode = K, A.isConcurrentMode = ie, A.isContextConsumer =
    G, A.isContextProvider = H, A.isElement = ce, A.isForwardRef = v, A.isFragment = q, A.isLazy = se, A.isMemo = de, A.isPortal = ae, A.isProfiler =
    Te, A.isStrictMode = Ae, A.isSuspense = Re, A.isSuspenseList = pr, A.isValidElementType = d, A.typeOf = E;
  }()), A;
}
c(mo, "requireReactIs_development");
var Vt;
function po() {
  return Vt || (Vt = 1, process.env.NODE_ENV === "production" ? Le.exports = fo() : Le.exports = mo()), Le.exports;
}
c(po, "requireReactIs");
var cn = po(), go = /* @__PURE__ */ on(cn), ho = /* @__PURE__ */ Zt({
  __proto__: null,
  default: go
}, [cn]), yo = [
  "isAsyncMode",
  "isConcurrentMode",
  "isContextConsumer",
  "isContextProvider",
  "isElement",
  "isForwardRef",
  "isFragment",
  "isLazy",
  "isMemo",
  "isPortal",
  "isProfiler",
  "isStrictMode",
  "isSuspense",
  "isSuspenseList",
  "isValidElementType"
], me = Object.fromEntries(yo.map((e) => [e, (t) => ho[e](t) || ao[e](t)]));
function un(e, t = []) {
  if (Array.isArray(e))
    for (let n of e)
      un(n, t);
  else e != null && e !== !1 && e !== "" && t.push(e);
  return t;
}
c(un, "getChildren");
function qt(e) {
  let t = e.type;
  if (typeof t == "string")
    return t;
  if (typeof t == "function")
    return t.displayName || t.name || "Unknown";
  if (me.isFragment(e))
    return "React.Fragment";
  if (me.isSuspense(e))
    return "React.Suspense";
  if (typeof t == "object" && t !== null) {
    if (me.isContextProvider(e))
      return "Context.Provider";
    if (me.isContextConsumer(e))
      return "Context.Consumer";
    if (me.isForwardRef(e)) {
      if (t.displayName)
        return t.displayName;
      let n = t.render.displayName || t.render.name || "";
      return n === "" ? "ForwardRef" : `ForwardRef(${n})`;
    }
    if (me.isMemo(e)) {
      let n = t.displayName || t.type.displayName || t.type.name || "";
      return n === "" ? "Memo" : `Memo(${n})`;
    }
  }
  return "UNDEFINED";
}
c(qt, "getType");
function bo(e) {
  let { props: t } = e;
  return Object.keys(t).filter((n) => n !== "children" && t[n] !== void 0).sort();
}
c(bo, "getPropKeys$1");
var So = /* @__PURE__ */ c((e, t, n, r, o, i) => ++r > t.maxDepth ? ct(qt(e), t) : st(qt(e), ot(bo(e), e.props, t, n + t.indent, r, o, i), it(
un(e.props.children), t, n + t.indent, r, o, i), t, n), "serialize$1"), Eo = /* @__PURE__ */ c((e) => e != null && me.isElement(e), "test$1"),
_o = {
  serialize: So,
  test: Eo
}, To = typeof Symbol == "function" && Symbol.for ? Symbol.for("react.test.json") : 245830487;
function Co(e) {
  let { props: t } = e;
  return t ? Object.keys(t).filter((n) => t[n] !== void 0).sort() : [];
}
c(Co, "getPropKeys");
var Oo = /* @__PURE__ */ c((e, t, n, r, o, i) => ++r > t.maxDepth ? ct(e.type, t) : st(e.type, e.props ? ot(Co(e), e.props, t, n + t.indent,
r, o, i) : "", e.children ? it(e.children, t, n + t.indent, r, o, i) : "", t, n), "serialize"), $o = /* @__PURE__ */ c((e) => e && e.$$typeof ===
To, "test"), wo = {
  serialize: Oo,
  test: $o
}, ln = Object.prototype.toString, Ao = Date.prototype.toISOString, Ro = Error.prototype.toString, Kt = RegExp.prototype.toString;
function xe(e) {
  return typeof e.constructor == "function" && e.constructor.name || "Object";
}
c(xe, "getConstructorName");
function No(e) {
  return typeof window < "u" && e === window;
}
c(No, "isWindow");
var Po = /^Symbol\((.*)\)(.*)$/, Io = /\n/g, Fe = class extends Error {
  static {
    c(this, "PrettyFormatPluginError");
  }
  constructor(t, n) {
    super(t), this.stack = n, this.name = this.constructor.name;
  }
};
function Mo(e) {
  return e === "[object Array]" || e === "[object ArrayBuffer]" || e === "[object DataView]" || e === "[object Float32Array]" || e === "[obj\
ect Float64Array]" || e === "[object Int8Array]" || e === "[object Int16Array]" || e === "[object Int32Array]" || e === "[object Uint8Array]" ||
  e === "[object Uint8ClampedArray]" || e === "[object Uint16Array]" || e === "[object Uint32Array]";
}
c(Mo, "isToStringedArrayType");
function Lo(e) {
  return Object.is(e, -0) ? "-0" : String(e);
}
c(Lo, "printNumber");
function xo(e) {
  return `${e}n`;
}
c(xo, "printBigInt");
function Gt(e, t) {
  return t ? `[Function ${e.name || "anonymous"}]` : "[Function]";
}
c(Gt, "printFunction");
function Ht(e) {
  return String(e).replace(Po, "Symbol($1)");
}
c(Ht, "printSymbol");
function Jt(e) {
  return `[${Ro.call(e)}]`;
}
c(Jt, "printError");
function an(e, t, n, r) {
  if (e === !0 || e === !1)
    return `${e}`;
  if (e === void 0)
    return "undefined";
  if (e === null)
    return "null";
  let o = typeof e;
  if (o === "number")
    return Lo(e);
  if (o === "bigint")
    return xo(e);
  if (o === "string")
    return r ? `"${e.replaceAll(/"|\\/g, "\\$&")}"` : `"${e}"`;
  if (o === "function")
    return Gt(e, t);
  if (o === "symbol")
    return Ht(e);
  let i = ln.call(e);
  return i === "[object WeakMap]" ? "WeakMap {}" : i === "[object WeakSet]" ? "WeakSet {}" : i === "[object Function]" || i === "[object Gen\
eratorFunction]" ? Gt(e, t) : i === "[object Symbol]" ? Ht(e) : i === "[object Date]" ? Number.isNaN(+e) ? "Date { NaN }" : Ao.call(e) : i ===
  "[object Error]" ? Jt(e) : i === "[object RegExp]" ? n ? Kt.call(e).replaceAll(/[$()*+.?[\\\]^{|}]/g, "\\$&") : Kt.call(e) : e instanceof Error ?
  Jt(e) : null;
}
c(an, "printBasicValue");
function fn(e, t, n, r, o, i) {
  if (o.includes(e))
    return "[Circular]";
  o = [...o], o.push(e);
  let s = ++r > t.maxDepth, a = t.min;
  if (t.callToJSON && !s && e.toJSON && typeof e.toJSON == "function" && !i)
    return fe(e.toJSON(), t, n, r, o, !0);
  let l = ln.call(e);
  return l === "[object Arguments]" ? s ? "[Arguments]" : `${a ? "" : "Arguments "}[${De(e, t, n, r, o, fe)}]` : Mo(l) ? s ? `[${e.constructor.
  name}]` : `${a || !t.printBasicPrototype && e.constructor.name === "Array" ? "" : `${e.constructor.name} `}[${De(e, t, n, r, o, fe)}]` : l ===
  "[object Map]" ? s ? "[Map]" : `Map {${Oe(e.entries(), t, n, r, o, fe, " => ")}}` : l === "[object Set]" ? s ? "[Set]" : `Set {${nt(e.values(),
  t, n, r, o, fe)}}` : s || No(e) ? `[${xe(e)}]` : `${a || !t.printBasicPrototype && xe(e) === "Object" ? "" : `${xe(e)} `}{${rt(e, t, n, r,
  o, fe)}}`;
}
c(fn, "printComplexValue");
var Do = {
  test: /* @__PURE__ */ c((e) => e && e instanceof Error, "test"),
  serialize(e, t, n, r, o, i) {
    if (o.includes(e))
      return "[Circular]";
    o = [...o, e];
    let s = ++r > t.maxDepth, { message: a, cause: l, ...m } = e, p = {
      message: a,
      ...typeof l < "u" ? { cause: l } : {},
      ...e instanceof AggregateError ? { errors: e.errors } : {},
      ...m
    }, f = e.name !== "Error" ? e.name : xe(e);
    return s ? `[${f}]` : `${f} {${Oe(Object.entries(p).values(), t, n, r, o, i)}}`;
  }
};
function Fo(e) {
  return e.serialize != null;
}
c(Fo, "isNewPlugin");
function mn(e, t, n, r, o, i) {
  let s;
  try {
    s = Fo(e) ? e.serialize(t, n, r, o, i, fe) : e.print(t, (a) => fe(a, n, r, o, i), (a) => {
      let l = r + n.indent;
      return l + a.replaceAll(Io, `
${l}`);
    }, {
      edgeSpacing: n.spacingOuter,
      min: n.min,
      spacing: n.spacingInner
    }, n.colors);
  } catch (a) {
    throw new Fe(a.message, a.stack);
  }
  if (typeof s != "string")
    throw new TypeError(`pretty-format: Plugin must return type "string" but instead returned "${typeof s}".`);
  return s;
}
c(mn, "printPlugin");
function pn(e, t) {
  for (let n of e)
    try {
      if (n.test(t))
        return n;
    } catch (r) {
      throw new Fe(r.message, r.stack);
    }
  return null;
}
c(pn, "findPlugin");
function fe(e, t, n, r, o, i) {
  let s = pn(t.plugins, e);
  if (s !== null)
    return mn(s, e, t, n, r, o);
  let a = an(e, t.printFunctionName, t.escapeRegex, t.escapeString);
  return a !== null ? a : fn(e, t, n, r, o, i);
}
c(fe, "printer");
var ut = {
  comment: "gray",
  content: "reset",
  prop: "yellow",
  tag: "cyan",
  value: "green"
}, gn = Object.keys(ut), re = {
  callToJSON: !0,
  compareKeys: void 0,
  escapeRegex: !1,
  escapeString: !0,
  highlight: !1,
  indent: 2,
  maxDepth: Number.POSITIVE_INFINITY,
  maxWidth: Number.POSITIVE_INFINITY,
  min: !1,
  plugins: [],
  printBasicPrototype: !0,
  printFunctionName: !0,
  theme: ut
};
function jo(e) {
  for (let t of Object.keys(e))
    if (!Object.prototype.hasOwnProperty.call(re, t))
      throw new Error(`pretty-format: Unknown option "${t}".`);
  if (e.min && e.indent !== void 0 && e.indent !== 0)
    throw new Error('pretty-format: Options "min" and "indent" cannot be used together.');
}
c(jo, "validateOptions");
function Bo() {
  return gn.reduce((e, t) => {
    let n = ut[t], r = n && ne[n];
    if (r && typeof r.close == "string" && typeof r.open == "string")
      e[t] = r;
    else
      throw new Error(`pretty-format: Option "theme" has a key "${t}" whose value "${n}" is undefined in ansi-styles.`);
    return e;
  }, /* @__PURE__ */ Object.create(null));
}
c(Bo, "getColorsHighlight");
function ko() {
  return gn.reduce((e, t) => (e[t] = {
    close: "",
    open: ""
  }, e), /* @__PURE__ */ Object.create(null));
}
c(ko, "getColorsEmpty");
function hn(e) {
  return e?.printFunctionName ?? re.printFunctionName;
}
c(hn, "getPrintFunctionName");
function yn(e) {
  return e?.escapeRegex ?? re.escapeRegex;
}
c(yn, "getEscapeRegex");
function dn(e) {
  return e?.escapeString ?? re.escapeString;
}
c(dn, "getEscapeString");
function Xt(e) {
  return {
    callToJSON: e?.callToJSON ?? re.callToJSON,
    colors: e?.highlight ? Bo() : ko(),
    compareKeys: typeof e?.compareKeys == "function" || e?.compareKeys === null ? e.compareKeys : re.compareKeys,
    escapeRegex: yn(e),
    escapeString: dn(e),
    indent: e?.min ? "" : zo(e?.indent ?? re.indent),
    maxDepth: e?.maxDepth ?? re.maxDepth,
    maxWidth: e?.maxWidth ?? re.maxWidth,
    min: e?.min ?? re.min,
    plugins: e?.plugins ?? re.plugins,
    printBasicPrototype: e?.printBasicPrototype ?? !0,
    printFunctionName: hn(e),
    spacingInner: e?.min ? " " : `
`,
    spacingOuter: e?.min ? "" : `
`
  };
}
c(Xt, "getConfig");
function zo(e) {
  return Array.from({ length: e + 1 }).join(" ");
}
c(zo, "createIndent");
function Q(e, t) {
  if (t && (jo(t), t.plugins)) {
    let r = pn(t.plugins, e);
    if (r !== null)
      return mn(r, e, Xt(t), "", 0, []);
  }
  let n = an(e, hn(t), yn(t), dn(t));
  return n !== null ? n : fn(e, Xt(t), "", 0, []);
}
c(Q, "format");
var $e = {
  AsymmetricMatcher: Ar,
  DOMCollection: xr,
  DOMElement: Vr,
  Immutable: io,
  ReactElement: _o,
  ReactTestComponent: wo,
  Error: Do
};

// ../node_modules/loupe/lib/helpers.js
var bn = {
  bold: ["1", "22"],
  dim: ["2", "22"],
  italic: ["3", "23"],
  underline: ["4", "24"],
  // 5 & 6 are blinking
  inverse: ["7", "27"],
  hidden: ["8", "28"],
  strike: ["9", "29"],
  // 10-20 are fonts
  // 21-29 are resets for 1-9
  black: ["30", "39"],
  red: ["31", "39"],
  green: ["32", "39"],
  yellow: ["33", "39"],
  blue: ["34", "39"],
  magenta: ["35", "39"],
  cyan: ["36", "39"],
  white: ["37", "39"],
  brightblack: ["30;1", "39"],
  brightred: ["31;1", "39"],
  brightgreen: ["32;1", "39"],
  brightyellow: ["33;1", "39"],
  brightblue: ["34;1", "39"],
  brightmagenta: ["35;1", "39"],
  brightcyan: ["36;1", "39"],
  brightwhite: ["37;1", "39"],
  grey: ["90", "39"]
}, Yo = {
  special: "cyan",
  number: "yellow",
  bigint: "yellow",
  boolean: "yellow",
  undefined: "grey",
  null: "bold",
  string: "green",
  symbol: "green",
  date: "magenta",
  regexp: "red"
}, ue = "\u2026";
function Uo(e, t) {
  let n = bn[Yo[t]] || bn[t] || "";
  return n ? `\x1B[${n[0]}m${String(e)}\x1B[${n[1]}m` : String(e);
}
c(Uo, "colorise");
function Sn({
  showHidden: e = !1,
  depth: t = 2,
  colors: n = !1,
  customInspect: r = !0,
  showProxy: o = !1,
  maxArrayLength: i = 1 / 0,
  breakLength: s = 1 / 0,
  seen: a = [],
  // eslint-disable-next-line no-shadow
  truncate: l = 1 / 0,
  stylize: m = String
} = {}, p) {
  let f = {
    showHidden: !!e,
    depth: Number(t),
    colors: !!n,
    customInspect: !!r,
    showProxy: !!o,
    maxArrayLength: Number(i),
    breakLength: Number(s),
    truncate: Number(l),
    seen: a,
    inspect: p,
    stylize: m
  };
  return f.colors && (f.stylize = Uo), f;
}
c(Sn, "normaliseOptions");
function Wo(e) {
  return e >= "\uD800" && e <= "\uDBFF";
}
c(Wo, "isHighSurrogate");
function Y(e, t, n = ue) {
  e = String(e);
  let r = n.length, o = e.length;
  if (r > t && o > r)
    return n;
  if (o > t && o > r) {
    let i = t - r;
    return i > 0 && Wo(e[i - 1]) && (i = i - 1), `${e.slice(0, i)}${n}`;
  }
  return e;
}
c(Y, "truncate");
function j(e, t, n, r = ", ") {
  n = n || t.inspect;
  let o = e.length;
  if (o === 0)
    return "";
  let i = t.truncate, s = "", a = "", l = "";
  for (let m = 0; m < o; m += 1) {
    let p = m + 1 === e.length, f = m + 2 === e.length;
    l = `${ue}(${e.length - m})`;
    let b = e[m];
    t.truncate = i - s.length - (p ? 0 : r.length);
    let y = a || n(b, t) + (p ? "" : r), g = s.length + y.length, u = g + l.length;
    if (p && g > i && s.length + l.length <= i || !p && !f && u > i || (a = p ? "" : n(e[m + 1], t) + (f ? "" : r), !p && f && u > i && g + a.
    length > i))
      break;
    if (s += y, !p && !f && g + a.length >= i) {
      l = `${ue}(${e.length - m - 1})`;
      break;
    }
    l = "";
  }
  return `${s}${l}`;
}
c(j, "inspectList");
function Vo(e) {
  return e.match(/^[a-zA-Z_][a-zA-Z_0-9]*$/) ? e : JSON.stringify(e).replace(/'/g, "\\'").replace(/\\"/g, '"').replace(/(^"|"$)/g, "'");
}
c(Vo, "quoteComplexKey");
function le([e, t], n) {
  return n.truncate -= 2, typeof e == "string" ? e = Vo(e) : typeof e != "number" && (e = `[${n.inspect(e, n)}]`), n.truncate -= e.length, t =
  n.inspect(t, n), `${e}: ${t}`;
}
c(le, "inspectProperty");

// ../node_modules/loupe/lib/array.js
function lt(e, t) {
  let n = Object.keys(e).slice(e.length);
  if (!e.length && !n.length)
    return "[]";
  t.truncate -= 4;
  let r = j(e, t);
  t.truncate -= r.length;
  let o = "";
  return n.length && (o = j(n.map((i) => [i, e[i]]), t, le)), `[ ${r}${o ? `, ${o}` : ""} ]`;
}
c(lt, "inspectArray");

// ../node_modules/loupe/lib/typedarray.js
var qo = /* @__PURE__ */ c((e) => typeof Buffer == "function" && e instanceof Buffer ? "Buffer" : e[Symbol.toStringTag] ? e[Symbol.toStringTag] :
e.constructor.name, "getArrayName");
function oe(e, t) {
  let n = qo(e);
  t.truncate -= n.length + 4;
  let r = Object.keys(e).slice(e.length);
  if (!e.length && !r.length)
    return `${n}[]`;
  let o = "";
  for (let s = 0; s < e.length; s++) {
    let a = `${t.stylize(Y(e[s], t.truncate), "number")}${s === e.length - 1 ? "" : ", "}`;
    if (t.truncate -= a.length, e[s] !== e.length && t.truncate <= 3) {
      o += `${ue}(${e.length - e[s] + 1})`;
      break;
    }
    o += a;
  }
  let i = "";
  return r.length && (i = j(r.map((s) => [s, e[s]]), t, le)), `${n}[ ${o}${i ? `, ${i}` : ""} ]`;
}
c(oe, "inspectTypedArray");

// ../node_modules/loupe/lib/date.js
function at(e, t) {
  let n = e.toJSON();
  if (n === null)
    return "Invalid Date";
  let r = n.split("T"), o = r[0];
  return t.stylize(`${o}T${Y(r[1], t.truncate - o.length - 1)}`, "date");
}
c(at, "inspectDate");

// ../node_modules/loupe/lib/function.js
function Be(e, t) {
  let n = e[Symbol.toStringTag] || "Function", r = e.name;
  return r ? t.stylize(`[${n} ${Y(r, t.truncate - 11)}]`, "special") : t.stylize(`[${n}]`, "special");
}
c(Be, "inspectFunction");

// ../node_modules/loupe/lib/map.js
function Ko([e, t], n) {
  return n.truncate -= 4, e = n.inspect(e, n), n.truncate -= e.length, t = n.inspect(t, n), `${e} => ${t}`;
}
c(Ko, "inspectMapEntry");
function Go(e) {
  let t = [];
  return e.forEach((n, r) => {
    t.push([r, n]);
  }), t;
}
c(Go, "mapToEntries");
function ft(e, t) {
  return e.size === 0 ? "Map{}" : (t.truncate -= 7, `Map{ ${j(Go(e), t, Ko)} }`);
}
c(ft, "inspectMap");

// ../node_modules/loupe/lib/number.js
var Ho = Number.isNaN || ((e) => e !== e);
function ke(e, t) {
  return Ho(e) ? t.stylize("NaN", "number") : e === 1 / 0 ? t.stylize("Infinity", "number") : e === -1 / 0 ? t.stylize("-Infinity", "number") :
  e === 0 ? t.stylize(1 / e === 1 / 0 ? "+0" : "-0", "number") : t.stylize(Y(String(e), t.truncate), "number");
}
c(ke, "inspectNumber");

// ../node_modules/loupe/lib/bigint.js
function ze(e, t) {
  let n = Y(e.toString(), t.truncate - 1);
  return n !== ue && (n += "n"), t.stylize(n, "bigint");
}
c(ze, "inspectBigInt");

// ../node_modules/loupe/lib/regexp.js
function mt(e, t) {
  let n = e.toString().split("/")[2], r = t.truncate - (2 + n.length), o = e.source;
  return t.stylize(`/${Y(o, r)}/${n}`, "regexp");
}
c(mt, "inspectRegExp");

// ../node_modules/loupe/lib/set.js
function Jo(e) {
  let t = [];
  return e.forEach((n) => {
    t.push(n);
  }), t;
}
c(Jo, "arrayFromSet");
function pt(e, t) {
  return e.size === 0 ? "Set{}" : (t.truncate -= 7, `Set{ ${j(Jo(e), t)} }`);
}
c(pt, "inspectSet");

// ../node_modules/loupe/lib/string.js
var En = new RegExp("['\\u0000-\\u001f\\u007f-\\u009f\\u00ad\\u0600-\\u0604\\u070f\\u17b4\\u17b5\\u200c-\\u200f\\u2028-\\u202f\\u2060-\\u206f\\ufeff\\ufff0-\\u\
ffff]", "g"), Xo = {
  "\b": "\\b",
  "	": "\\t",
  "\n": "\\n",
  "\f": "\\f",
  "\r": "\\r",
  "'": "\\'",
  "\\": "\\\\"
}, Zo = 16, vo = 4;
function Qo(e) {
  return Xo[e] || `\\u${`0000${e.charCodeAt(0).toString(Zo)}`.slice(-vo)}`;
}
c(Qo, "escape");
function Ye(e, t) {
  return En.test(e) && (e = e.replace(En, Qo)), t.stylize(`'${Y(e, t.truncate - 2)}'`, "string");
}
c(Ye, "inspectString");

// ../node_modules/loupe/lib/symbol.js
function Ue(e) {
  return "description" in Symbol.prototype ? e.description ? `Symbol(${e.description})` : "Symbol()" : e.toString();
}
c(Ue, "inspectSymbol");

// ../node_modules/loupe/lib/promise.js
var _n = /* @__PURE__ */ c(() => "Promise{\u2026}", "getPromiseValue");
try {
  let { getPromiseDetails: e, kPending: t, kRejected: n } = process.binding("util");
  Array.isArray(e(Promise.resolve())) && (_n = /* @__PURE__ */ c((r, o) => {
    let [i, s] = e(r);
    return i === t ? "Promise{<pending>}" : `Promise${i === n ? "!" : ""}{${o.inspect(s, o)}}`;
  }, "getPromiseValue"));
} catch {
}
var Tn = _n;

// ../node_modules/loupe/lib/object.js
function pe(e, t) {
  let n = Object.getOwnPropertyNames(e), r = Object.getOwnPropertySymbols ? Object.getOwnPropertySymbols(e) : [];
  if (n.length === 0 && r.length === 0)
    return "{}";
  if (t.truncate -= 4, t.seen = t.seen || [], t.seen.includes(e))
    return "[Circular]";
  t.seen.push(e);
  let o = j(n.map((a) => [a, e[a]]), t, le), i = j(r.map((a) => [a, e[a]]), t, le);
  t.seen.pop();
  let s = "";
  return o && i && (s = ", "), `{ ${o}${s}${i} }`;
}
c(pe, "inspectObject");

// ../node_modules/loupe/lib/class.js
var gt = typeof Symbol < "u" && Symbol.toStringTag ? Symbol.toStringTag : !1;
function ht(e, t) {
  let n = "";
  return gt && gt in e && (n = e[gt]), n = n || e.constructor.name, (!n || n === "_class") && (n = "<Anonymous Class>"), t.truncate -= n.length,
  `${n}${pe(e, t)}`;
}
c(ht, "inspectClass");

// ../node_modules/loupe/lib/arguments.js
function yt(e, t) {
  return e.length === 0 ? "Arguments[]" : (t.truncate -= 13, `Arguments[ ${j(e, t)} ]`);
}
c(yt, "inspectArguments");

// ../node_modules/loupe/lib/error.js
var ei = [
  "stack",
  "line",
  "column",
  "name",
  "message",
  "fileName",
  "lineNumber",
  "columnNumber",
  "number",
  "description",
  "cause"
];
function dt(e, t) {
  let n = Object.getOwnPropertyNames(e).filter((s) => ei.indexOf(s) === -1), r = e.name;
  t.truncate -= r.length;
  let o = "";
  if (typeof e.message == "string" ? o = Y(e.message, t.truncate) : n.unshift("message"), o = o ? `: ${o}` : "", t.truncate -= o.length + 5,
  t.seen = t.seen || [], t.seen.includes(e))
    return "[Circular]";
  t.seen.push(e);
  let i = j(n.map((s) => [s, e[s]]), t, le);
  return `${r}${o}${i ? ` { ${i} }` : ""}`;
}
c(dt, "inspectObject");

// ../node_modules/loupe/lib/html.js
function ti([e, t], n) {
  return n.truncate -= 3, t ? `${n.stylize(String(e), "yellow")}=${n.stylize(`"${t}"`, "string")}` : `${n.stylize(String(e), "yellow")}`;
}
c(ti, "inspectAttribute");
function We(e, t) {
  return j(e, t, Ve, `
`);
}
c(We, "inspectHTMLCollection");
function Ve(e, t) {
  let n = e.getAttributeNames(), r = e.tagName.toLowerCase(), o = t.stylize(`<${r}`, "special"), i = t.stylize(">", "special"), s = t.stylize(
  `</${r}>`, "special");
  t.truncate -= r.length * 2 + 5;
  let a = "";
  n.length > 0 && (a += " ", a += j(n.map((p) => [p, e.getAttribute(p)]), t, ti, " ")), t.truncate -= a.length;
  let l = t.truncate, m = We(e.children, t);
  return m && m.length > l && (m = `${ue}(${e.children.length})`), `${o}${a}${i}${m}${s}`;
}
c(Ve, "inspectHTML");

// ../node_modules/loupe/lib/index.js
var ni = typeof Symbol == "function" && typeof Symbol.for == "function", bt = ni ? Symbol.for("chai/inspect") : "@@chai/inspect", Se = !1;
try {
  let e = require("util");
  Se = e.inspect ? e.inspect.custom : !1;
} catch {
  Se = !1;
}
var Cn = /* @__PURE__ */ new WeakMap(), On = {}, $n = {
  undefined: /* @__PURE__ */ c((e, t) => t.stylize("undefined", "undefined"), "undefined"),
  null: /* @__PURE__ */ c((e, t) => t.stylize("null", "null"), "null"),
  boolean: /* @__PURE__ */ c((e, t) => t.stylize(String(e), "boolean"), "boolean"),
  Boolean: /* @__PURE__ */ c((e, t) => t.stylize(String(e), "boolean"), "Boolean"),
  number: ke,
  Number: ke,
  bigint: ze,
  BigInt: ze,
  string: Ye,
  String: Ye,
  function: Be,
  Function: Be,
  symbol: Ue,
  // A Symbol polyfill will return `Symbol` not `symbol` from typedetect
  Symbol: Ue,
  Array: lt,
  Date: at,
  Map: ft,
  Set: pt,
  RegExp: mt,
  Promise: Tn,
  // WeakSet, WeakMap are totally opaque to us
  WeakSet: /* @__PURE__ */ c((e, t) => t.stylize("WeakSet{\u2026}", "special"), "WeakSet"),
  WeakMap: /* @__PURE__ */ c((e, t) => t.stylize("WeakMap{\u2026}", "special"), "WeakMap"),
  Arguments: yt,
  Int8Array: oe,
  Uint8Array: oe,
  Uint8ClampedArray: oe,
  Int16Array: oe,
  Uint16Array: oe,
  Int32Array: oe,
  Uint32Array: oe,
  Float32Array: oe,
  Float64Array: oe,
  Generator: /* @__PURE__ */ c(() => "", "Generator"),
  DataView: /* @__PURE__ */ c(() => "", "DataView"),
  ArrayBuffer: /* @__PURE__ */ c(() => "", "ArrayBuffer"),
  Error: dt,
  HTMLCollection: We,
  NodeList: We
}, ri = /* @__PURE__ */ c((e, t, n) => bt in e && typeof e[bt] == "function" ? e[bt](t) : Se && Se in e && typeof e[Se] == "function" ? e[Se](
t.depth, t) : "inspect" in e && typeof e.inspect == "function" ? e.inspect(t.depth, t) : "constructor" in e && Cn.has(e.constructor) ? Cn.get(
e.constructor)(e, t) : On[n] ? On[n](e, t) : "", "inspectCustom"), oi = Object.prototype.toString;
function qe(e, t = {}) {
  let n = Sn(t, qe), { customInspect: r } = n, o = e === null ? "null" : typeof e;
  if (o === "object" && (o = oi.call(e).slice(8, -1)), o in $n)
    return $n[o](e, n);
  if (r && e) {
    let s = ri(e, n, o);
    if (s)
      return typeof s == "string" ? s : qe(s, n);
  }
  let i = e ? Object.getPrototypeOf(e) : !1;
  return i === Object.prototype || i === null ? pe(e, n) : e && typeof HTMLElement == "function" && e instanceof HTMLElement ? Ve(e, n) : "c\
onstructor" in e ? e.constructor !== Object ? ht(e, n) : pe(e, n) : e === Object(e) ? pe(e, n) : n.stylize(String(e), o);
}
c(qe, "inspect");

// ../node_modules/@vitest/utils/dist/chunk-_commonjsHelpers.js
var { AsymmetricMatcher: si, DOMCollection: ci, DOMElement: ui, Immutable: li, ReactElement: ai, ReactTestComponent: fi } = $e, wn = [
  fi,
  ai,
  ui,
  ci,
  li,
  si
];
function ge(e, t = 10, { maxLength: n, ...r } = {}) {
  let o = n ?? 1e4, i;
  try {
    i = Q(e, {
      maxDepth: t,
      escapeString: !1,
      plugins: wn,
      ...r
    });
  } catch {
    i = Q(e, {
      callToJSON: !1,
      maxDepth: t,
      escapeString: !1,
      plugins: wn,
      ...r
    });
  }
  return i.length >= o && t > 1 ? ge(e, Math.floor(Math.min(t, Number.MAX_SAFE_INTEGER) / 2), {
    maxLength: n,
    ...r
  }) : i;
}
c(ge, "stringify");
var mi = /%[sdjifoOc%]/g;
function An(...e) {
  if (typeof e[0] != "string") {
    let i = [];
    for (let s = 0; s < e.length; s++)
      i.push(we(e[s], {
        depth: 0,
        colors: !1
      }));
    return i.join(" ");
  }
  let t = e.length, n = 1, r = e[0], o = String(r).replace(mi, (i) => {
    if (i === "%%")
      return "%";
    if (n >= t)
      return i;
    switch (i) {
      case "%s": {
        let s = e[n++];
        return typeof s == "bigint" ? `${s.toString()}n` : typeof s == "number" && s === 0 && 1 / s < 0 ? "-0" : typeof s == "object" && s !==
        null ? typeof s.toString == "function" && s.toString !== Object.prototype.toString ? s.toString() : we(s, {
          depth: 0,
          colors: !1
        }) : String(s);
      }
      case "%d": {
        let s = e[n++];
        return typeof s == "bigint" ? `${s.toString()}n` : Number(s).toString();
      }
      case "%i": {
        let s = e[n++];
        return typeof s == "bigint" ? `${s.toString()}n` : Number.parseInt(String(s)).toString();
      }
      case "%f":
        return Number.parseFloat(String(e[n++])).toString();
      case "%o":
        return we(e[n++], {
          showHidden: !0,
          showProxy: !0
        });
      case "%O":
        return we(e[n++]);
      case "%c":
        return n++, "";
      case "%j":
        try {
          return JSON.stringify(e[n++]);
        } catch (s) {
          let a = s.message;
          if (a.includes("circular structure") || a.includes("cyclic structures") || a.includes("cyclic object"))
            return "[Circular]";
          throw s;
        }
      default:
        return i;
    }
  });
  for (let i = e[n]; n < t; i = e[++n])
    i === null || typeof i != "object" ? o += ` ${i}` : o += ` ${we(i)}`;
  return o;
}
c(An, "format");
function we(e, t = {}) {
  return t.truncate === 0 && (t.truncate = Number.POSITIVE_INFINITY), qe(e, t);
}
c(we, "inspect");
function Rn(e) {
  return e && e.__esModule && Object.prototype.hasOwnProperty.call(e, "default") ? e.default : e;
}
c(Rn, "getDefaultExportFromCjs");

// ../node_modules/@vitest/utils/dist/helpers.js
function pi(e) {
  return e === Object.prototype || e === Function.prototype || e === RegExp.prototype;
}
c(pi, "isFinalObj");
function Ke(e) {
  return Object.prototype.toString.apply(e).slice(8, -1);
}
c(Ke, "getType");
function gi(e, t) {
  let n = typeof t == "function" ? t : (r) => t.add(r);
  Object.getOwnPropertyNames(e).forEach(n), Object.getOwnPropertySymbols(e).forEach(n);
}
c(gi, "collectOwnProperties");
function Et(e) {
  let t = /* @__PURE__ */ new Set();
  return pi(e) ? [] : (gi(e, t), Array.from(t));
}
c(Et, "getOwnProperties");
var Nn = { forceWritable: !1 };
function _t(e, t = Nn) {
  return St(e, /* @__PURE__ */ new WeakMap(), t);
}
c(_t, "deepClone");
function St(e, t, n = Nn) {
  let r, o;
  if (t.has(e))
    return t.get(e);
  if (Array.isArray(e)) {
    for (o = Array.from({ length: r = e.length }), t.set(e, o); r--; )
      o[r] = St(e[r], t, n);
    return o;
  }
  if (Object.prototype.toString.call(e) === "[object Object]") {
    o = Object.create(Object.getPrototypeOf(e)), t.set(e, o);
    let i = Et(e);
    for (let s of i) {
      let a = Object.getOwnPropertyDescriptor(e, s);
      if (!a)
        continue;
      let l = St(e[s], t, n);
      n.forceWritable ? Object.defineProperty(o, s, {
        enumerable: a.enumerable,
        configurable: !0,
        writable: !0,
        value: l
      }) : "get" in a ? Object.defineProperty(o, s, {
        ...a,
        get() {
          return l;
        }
      }) : Object.defineProperty(o, s, {
        ...a,
        value: l
      });
    }
    return o;
  }
  return e;
}
c(St, "clone");

// ../node_modules/@vitest/utils/dist/diff.js
var U = -1, k = 1, x = 0, I = class {
  static {
    c(this, "Diff");
  }
  0;
  1;
  constructor(t, n) {
    this[0] = t, this[1] = n;
  }
};
function hi(e, t) {
  if (!e || !t || e.charAt(0) !== t.charAt(0))
    return 0;
  let n = 0, r = Math.min(e.length, t.length), o = r, i = 0;
  for (; n < o; )
    e.substring(i, o) === t.substring(i, o) ? (n = o, i = n) : r = o, o = Math.floor((r - n) / 2 + n);
  return o;
}
c(hi, "diff_commonPrefix");
function qn(e, t) {
  if (!e || !t || e.charAt(e.length - 1) !== t.charAt(t.length - 1))
    return 0;
  let n = 0, r = Math.min(e.length, t.length), o = r, i = 0;
  for (; n < o; )
    e.substring(e.length - o, e.length - i) === t.substring(t.length - o, t.length - i) ? (n = o, i = n) : r = o, o = Math.floor((r - n) / 2 +
    n);
  return o;
}
c(qn, "diff_commonSuffix");
function Pn(e, t) {
  let n = e.length, r = t.length;
  if (n === 0 || r === 0)
    return 0;
  n > r ? e = e.substring(n - r) : n < r && (t = t.substring(0, n));
  let o = Math.min(n, r);
  if (e === t)
    return o;
  let i = 0, s = 1;
  for (; ; ) {
    let a = e.substring(o - s), l = t.indexOf(a);
    if (l === -1)
      return i;
    s += l, (l === 0 || e.substring(o - s) === t.substring(0, s)) && (i = s, s++);
  }
}
c(Pn, "diff_commonOverlap_");
function yi(e) {
  let t = !1, n = [], r = 0, o = null, i = 0, s = 0, a = 0, l = 0, m = 0;
  for (; i < e.length; )
    e[i][0] === x ? (n[r++] = i, s = l, a = m, l = 0, m = 0, o = e[i][1]) : (e[i][0] === k ? l += e[i][1].length : m += e[i][1].length, o &&
    o.length <= Math.max(s, a) && o.length <= Math.max(l, m) && (e.splice(n[r - 1], 0, new I(U, o)), e[n[r - 1] + 1][0] = k, r--, r--, i = r >
    0 ? n[r - 1] : -1, s = 0, a = 0, l = 0, m = 0, o = null, t = !0)), i++;
  for (t && Kn(e), Si(e), i = 1; i < e.length; ) {
    if (e[i - 1][0] === U && e[i][0] === k) {
      let p = e[i - 1][1], f = e[i][1], b = Pn(p, f), y = Pn(f, p);
      b >= y ? (b >= p.length / 2 || b >= f.length / 2) && (e.splice(i, 0, new I(x, f.substring(0, b))), e[i - 1][1] = p.substring(0, p.length -
      b), e[i + 1][1] = f.substring(b), i++) : (y >= p.length / 2 || y >= f.length / 2) && (e.splice(i, 0, new I(x, p.substring(0, y))), e[i -
      1][0] = k, e[i - 1][1] = f.substring(0, f.length - y), e[i + 1][0] = U, e[i + 1][1] = p.substring(y), i++), i++;
    }
    i++;
  }
}
c(yi, "diff_cleanupSemantic");
var In = /[^a-z0-9]/i, Mn = /\s/, Ln = /[\r\n]/, di = /\n\r?\n$/, bi = /^\r?\n\r?\n/;
function Si(e) {
  let t = 1;
  for (; t < e.length - 1; ) {
    if (e[t - 1][0] === x && e[t + 1][0] === x) {
      let n = e[t - 1][1], r = e[t][1], o = e[t + 1][1], i = qn(n, r);
      if (i) {
        let p = r.substring(r.length - i);
        n = n.substring(0, n.length - i), r = p + r.substring(0, r.length - i), o = p + o;
      }
      let s = n, a = r, l = o, m = Ge(n, r) + Ge(r, o);
      for (; r.charAt(0) === o.charAt(0); ) {
        n += r.charAt(0), r = r.substring(1) + o.charAt(0), o = o.substring(1);
        let p = Ge(n, r) + Ge(r, o);
        p >= m && (m = p, s = n, a = r, l = o);
      }
      e[t - 1][1] !== s && (s ? e[t - 1][1] = s : (e.splice(t - 1, 1), t--), e[t][1] = a, l ? e[t + 1][1] = l : (e.splice(t + 1, 1), t--));
    }
    t++;
  }
}
c(Si, "diff_cleanupSemanticLossless");
function Kn(e) {
  e.push(new I(x, ""));
  let t = 0, n = 0, r = 0, o = "", i = "", s;
  for (; t < e.length; )
    switch (e[t][0]) {
      case k:
        r++, i += e[t][1], t++;
        break;
      case U:
        n++, o += e[t][1], t++;
        break;
      case x:
        n + r > 1 ? (n !== 0 && r !== 0 && (s = hi(i, o), s !== 0 && (t - n - r > 0 && e[t - n - r - 1][0] === x ? e[t - n - r - 1][1] += i.
        substring(0, s) : (e.splice(0, 0, new I(x, i.substring(0, s))), t++), i = i.substring(s), o = o.substring(s)), s = qn(i, o), s !== 0 &&
        (e[t][1] = i.substring(i.length - s) + e[t][1], i = i.substring(0, i.length - s), o = o.substring(0, o.length - s))), t -= n + r, e.
        splice(t, n + r), o.length && (e.splice(t, 0, new I(U, o)), t++), i.length && (e.splice(t, 0, new I(k, i)), t++), t++) : t !== 0 && e[t -
        1][0] === x ? (e[t - 1][1] += e[t][1], e.splice(t, 1)) : t++, r = 0, n = 0, o = "", i = "";
        break;
    }
  e[e.length - 1][1] === "" && e.pop();
  let a = !1;
  for (t = 1; t < e.length - 1; )
    e[t - 1][0] === x && e[t + 1][0] === x && (e[t][1].substring(e[t][1].length - e[t - 1][1].length) === e[t - 1][1] ? (e[t][1] = e[t - 1][1] +
    e[t][1].substring(0, e[t][1].length - e[t - 1][1].length), e[t + 1][1] = e[t - 1][1] + e[t + 1][1], e.splice(t - 1, 1), a = !0) : e[t][1].
    substring(0, e[t + 1][1].length) === e[t + 1][1] && (e[t - 1][1] += e[t + 1][1], e[t][1] = e[t][1].substring(e[t + 1][1].length) + e[t +
    1][1], e.splice(t + 1, 1), a = !0)), t++;
  a && Kn(e);
}
c(Kn, "diff_cleanupMerge");
function Ge(e, t) {
  if (!e || !t)
    return 6;
  let n = e.charAt(e.length - 1), r = t.charAt(0), o = n.match(In), i = r.match(In), s = o && n.match(Mn), a = i && r.match(Mn), l = s && n.
  match(Ln), m = a && r.match(Ln), p = l && e.match(di), f = m && t.match(bi);
  return p || f ? 5 : l || m ? 4 : o && !s && a ? 3 : s || a ? 2 : o || i ? 1 : 0;
}
c(Ge, "diff_cleanupSemanticScore_");
var Gn = "Compared values have no visual difference.", Ei = "Compared values serialize to the same structure.\nPrinting internal object struc\
ture without calling `toJSON` instead.", He = {}, xn;
function _i() {
  if (xn) return He;
  xn = 1, Object.defineProperty(He, "__esModule", {
    value: !0
  }), He.default = b;
  let e = "diff-sequences", t = 0, n = /* @__PURE__ */ c((y, g, u, h, S) => {
    let _ = 0;
    for (; y < g && u < h && S(y, u); )
      y += 1, u += 1, _ += 1;
    return _;
  }, "countCommonItemsF"), r = /* @__PURE__ */ c((y, g, u, h, S) => {
    let _ = 0;
    for (; y <= g && u <= h && S(g, h); )
      g -= 1, h -= 1, _ += 1;
    return _;
  }, "countCommonItemsR"), o = /* @__PURE__ */ c((y, g, u, h, S, _, O) => {
    let d = 0, E = -y, $ = _[d], T = $;
    _[d] += n(
      $ + 1,
      g,
      h + $ - E + 1,
      u,
      S
    );
    let R = y < O ? y : O;
    for (d += 1, E += 2; d <= R; d += 1, E += 2) {
      if (d !== y && T < _[d])
        $ = _[d];
      else if ($ = T + 1, g <= $)
        return d - 1;
      T = _[d], _[d] = $ + n($ + 1, g, h + $ - E + 1, u, S);
    }
    return O;
  }, "extendPathsF"), i = /* @__PURE__ */ c((y, g, u, h, S, _, O) => {
    let d = 0, E = y, $ = _[d], T = $;
    _[d] -= r(
      g,
      $ - 1,
      u,
      h + $ - E - 1,
      S
    );
    let R = y < O ? y : O;
    for (d += 1, E -= 2; d <= R; d += 1, E -= 2) {
      if (d !== y && _[d] < T)
        $ = _[d];
      else if ($ = T - 1, $ < g)
        return d - 1;
      T = _[d], _[d] = $ - r(
        g,
        $ - 1,
        u,
        h + $ - E - 1,
        S
      );
    }
    return O;
  }, "extendPathsR"), s = /* @__PURE__ */ c((y, g, u, h, S, _, O, d, E, $, T) => {
    let R = h - g, J = u - g, L = S - h - J, z = -L - (y - 1), X = -L + (y - 1), V = t, M = y < d ? y : d;
    for (let D = 0, F = -y; D <= M; D += 1, F += 2) {
      let Z = D === 0 || D !== y && V < O[D], B = Z ? O[D] : V, K = Z ? B : B + 1, ie = R + K - F, G = n(
        K + 1,
        u,
        ie + 1,
        S,
        _
      ), H = K + G;
      if (V = O[D], O[D] = H, z <= F && F <= X) {
        let ce = (y - 1 - (F + L)) / 2;
        if (ce <= $ && E[ce] - 1 <= H) {
          let v = R + B - (Z ? F + 1 : F - 1), q = r(
            g,
            B,
            h,
            v,
            _
          ), se = B - q, de = v - q, ae = se + 1, Te = de + 1;
          T.nChangePreceding = y - 1, y - 1 === ae + Te - g - h ? (T.aEndPreceding = g, T.bEndPreceding = h) : (T.aEndPreceding = ae, T.bEndPreceding =
          Te), T.nCommonPreceding = q, q !== 0 && (T.aCommonPreceding = ae, T.bCommonPreceding = Te), T.nCommonFollowing = G, G !== 0 && (T.
          aCommonFollowing = K + 1, T.bCommonFollowing = ie + 1);
          let Ae = H + 1, Re = ie + G + 1;
          return T.nChangeFollowing = y - 1, y - 1 === u + S - Ae - Re ? (T.aStartFollowing = u, T.bStartFollowing = S) : (T.aStartFollowing =
          Ae, T.bStartFollowing = Re), !0;
        }
      }
    }
    return !1;
  }, "extendOverlappablePathsF"), a = /* @__PURE__ */ c((y, g, u, h, S, _, O, d, E, $, T) => {
    let R = S - u, J = u - g, L = S - h - J, z = L - y, X = L + y, V = t, M = y < $ ? y : $;
    for (let D = 0, F = y; D <= M; D += 1, F -= 2) {
      let Z = D === 0 || D !== y && E[D] < V, B = Z ? E[D] : V, K = Z ? B : B - 1, ie = R + K - F, G = r(
        g,
        K - 1,
        h,
        ie - 1,
        _
      ), H = K - G;
      if (V = E[D], E[D] = H, z <= F && F <= X) {
        let ce = (y + (F - L)) / 2;
        if (ce <= d && H - 1 <= O[ce]) {
          let v = ie - G;
          if (T.nChangePreceding = y, y === H + v - g - h ? (T.aEndPreceding = g, T.bEndPreceding = h) : (T.aEndPreceding = H, T.bEndPreceding =
          v), T.nCommonPreceding = G, G !== 0 && (T.aCommonPreceding = H, T.bCommonPreceding = v), T.nChangeFollowing = y - 1, y === 1)
            T.nCommonFollowing = 0, T.aStartFollowing = u, T.bStartFollowing = S;
          else {
            let q = R + B - (Z ? F - 1 : F + 1), se = n(
              B,
              u,
              q,
              S,
              _
            );
            T.nCommonFollowing = se, se !== 0 && (T.aCommonFollowing = B, T.bCommonFollowing = q);
            let de = B + se, ae = q + se;
            y - 1 === u + S - de - ae ? (T.aStartFollowing = u, T.bStartFollowing = S) : (T.aStartFollowing = de, T.bStartFollowing = ae);
          }
          return !0;
        }
      }
    }
    return !1;
  }, "extendOverlappablePathsR"), l = /* @__PURE__ */ c((y, g, u, h, S, _, O, d, E) => {
    let $ = h - g, T = S - u, R = u - g, J = S - h, te = J - R, L = R, z = R;
    if (O[0] = g - 1, d[0] = u, te % 2 === 0) {
      let X = (y || te) / 2, V = (R + J) / 2;
      for (let M = 1; M <= V; M += 1)
        if (L = o(M, u, S, $, _, O, L), M < X)
          z = i(M, g, h, T, _, d, z);
        else if (
          // If a reverse path overlaps a forward path in the same diagonal,
          // return a division of the index intervals at the middle change.
          a(
            M,
            g,
            u,
            h,
            S,
            _,
            O,
            L,
            d,
            z,
            E
          )
        )
          return;
    } else {
      let X = ((y || te) + 1) / 2, V = (R + J + 1) / 2, M = 1;
      for (L = o(M, u, S, $, _, O, L), M += 1; M <= V; M += 1)
        if (z = i(
          M - 1,
          g,
          h,
          T,
          _,
          d,
          z
        ), M < X)
          L = o(M, u, S, $, _, O, L);
        else if (
          // If a forward path overlaps a reverse path in the same diagonal,
          // return a division of the index intervals at the middle change.
          s(
            M,
            g,
            u,
            h,
            S,
            _,
            O,
            L,
            d,
            z,
            E
          )
        )
          return;
    }
    throw new Error(
      `${e}: no overlap aStart=${g} aEnd=${u} bStart=${h} bEnd=${S}`
    );
  }, "divide"), m = /* @__PURE__ */ c((y, g, u, h, S, _, O, d, E, $) => {
    if (S - h < u - g) {
      if (_ = !_, _ && O.length === 1) {
        let { foundSubsequence: H, isCommon: ce } = O[0];
        O[1] = {
          foundSubsequence: /* @__PURE__ */ c((v, q, se) => {
            H(v, se, q);
          }, "foundSubsequence"),
          isCommon: /* @__PURE__ */ c((v, q) => ce(q, v), "isCommon")
        };
      }
      let ie = g, G = u;
      g = h, u = S, h = ie, S = G;
    }
    let { foundSubsequence: T, isCommon: R } = O[_ ? 1 : 0];
    l(
      y,
      g,
      u,
      h,
      S,
      R,
      d,
      E,
      $
    );
    let {
      nChangePreceding: J,
      aEndPreceding: te,
      bEndPreceding: L,
      nCommonPreceding: z,
      aCommonPreceding: X,
      bCommonPreceding: V,
      nCommonFollowing: M,
      aCommonFollowing: D,
      bCommonFollowing: F,
      nChangeFollowing: Z,
      aStartFollowing: B,
      bStartFollowing: K
    } = $;
    g < te && h < L && m(
      J,
      g,
      te,
      h,
      L,
      _,
      O,
      d,
      E,
      $
    ), z !== 0 && T(z, X, V), M !== 0 && T(M, D, F), B < u && K < S && m(
      Z,
      B,
      u,
      K,
      S,
      _,
      O,
      d,
      E,
      $
    );
  }, "findSubsequences"), p = /* @__PURE__ */ c((y, g) => {
    if (typeof g != "number")
      throw new TypeError(`${e}: ${y} typeof ${typeof g} is not a number`);
    if (!Number.isSafeInteger(g))
      throw new RangeError(`${e}: ${y} value ${g} is not a safe integer`);
    if (g < 0)
      throw new RangeError(`${e}: ${y} value ${g} is a negative integer`);
  }, "validateLength"), f = /* @__PURE__ */ c((y, g) => {
    let u = typeof g;
    if (u !== "function")
      throw new TypeError(`${e}: ${y} typeof ${u} is not a function`);
  }, "validateCallback");
  function b(y, g, u, h) {
    p("aLength", y), p("bLength", g), f("isCommon", u), f("foundSubsequence", h);
    let S = n(0, y, 0, g, u);
    if (S !== 0 && h(S, 0, 0), y !== S || g !== S) {
      let _ = S, O = S, d = r(
        _,
        y - 1,
        O,
        g - 1,
        u
      ), E = y - d, $ = g - d, T = S + d;
      y !== T && g !== T && m(
        0,
        _,
        E,
        O,
        $,
        !1,
        [
          {
            foundSubsequence: h,
            isCommon: u
          }
        ],
        [t],
        [t],
        {
          aCommonFollowing: t,
          aCommonPreceding: t,
          aEndPreceding: t,
          aStartFollowing: t,
          bCommonFollowing: t,
          bCommonPreceding: t,
          bEndPreceding: t,
          bStartFollowing: t,
          nChangeFollowing: t,
          nChangePreceding: t,
          nCommonFollowing: t,
          nCommonPreceding: t
        }
      ), d !== 0 && h(d, E, $);
    }
  }
  return c(b, "diffSequence"), He;
}
c(_i, "requireBuild");
var Ti = _i(), Hn = /* @__PURE__ */ Rn(Ti);
function Ci(e, t) {
  return e.replace(/\s+$/, (n) => t(n));
}
c(Ci, "formatTrailingSpaces");
function Rt(e, t, n, r, o, i) {
  return e.length !== 0 ? n(`${r} ${Ci(e, o)}`) : r !== " " ? n(r) : t && i.length !== 0 ? n(`${r} ${i}`) : "";
}
c(Rt, "printDiffLine");
function Jn(e, t, { aColor: n, aIndicator: r, changeLineTrailingSpaceColor: o, emptyFirstOrLastLinePlaceholder: i }) {
  return Rt(e, t, n, r, o, i);
}
c(Jn, "printDeleteLine");
function Xn(e, t, { bColor: n, bIndicator: r, changeLineTrailingSpaceColor: o, emptyFirstOrLastLinePlaceholder: i }) {
  return Rt(e, t, n, r, o, i);
}
c(Xn, "printInsertLine");
function Zn(e, t, { commonColor: n, commonIndicator: r, commonLineTrailingSpaceColor: o, emptyFirstOrLastLinePlaceholder: i }) {
  return Rt(e, t, n, r, o, i);
}
c(Zn, "printCommonLine");
function Dn(e, t, n, r, { patchColor: o }) {
  return o(`@@ -${e + 1},${t - e} +${n + 1},${r - n} @@`);
}
c(Dn, "createPatchMark");
function Oi(e, t) {
  let n = e.length, r = t.contextLines, o = r + r, i = n, s = !1, a = 0, l = 0;
  for (; l !== n; ) {
    let d = l;
    for (; l !== n && e[l][0] === x; )
      l += 1;
    if (d !== l)
      if (d === 0)
        l > r && (i -= l - r, s = !0);
      else if (l === n) {
        let E = l - d;
        E > r && (i -= E - r, s = !0);
      } else {
        let E = l - d;
        E > o && (i -= E - o, a += 1);
      }
    for (; l !== n && e[l][0] !== x; )
      l += 1;
  }
  let m = a !== 0 || s;
  a !== 0 ? i += a + 1 : s && (i += 1);
  let p = i - 1, f = [], b = 0;
  m && f.push("");
  let y = 0, g = 0, u = 0, h = 0, S = /* @__PURE__ */ c((d) => {
    let E = f.length;
    f.push(Zn(d, E === 0 || E === p, t)), u += 1, h += 1;
  }, "pushCommonLine"), _ = /* @__PURE__ */ c((d) => {
    let E = f.length;
    f.push(Jn(d, E === 0 || E === p, t)), u += 1;
  }, "pushDeleteLine"), O = /* @__PURE__ */ c((d) => {
    let E = f.length;
    f.push(Xn(d, E === 0 || E === p, t)), h += 1;
  }, "pushInsertLine");
  for (l = 0; l !== n; ) {
    let d = l;
    for (; l !== n && e[l][0] === x; )
      l += 1;
    if (d !== l)
      if (d === 0) {
        l > r && (d = l - r, y = d, g = d, u = y, h = g);
        for (let E = d; E !== l; E += 1)
          S(e[E][1]);
      } else if (l === n) {
        let E = l - d > r ? d + r : l;
        for (let $ = d; $ !== E; $ += 1)
          S(e[$][1]);
      } else {
        let E = l - d;
        if (E > o) {
          let $ = d + r;
          for (let R = d; R !== $; R += 1)
            S(e[R][1]);
          f[b] = Dn(y, u, g, h, t), b = f.length, f.push("");
          let T = E - o;
          y = u + T, g = h + T, u = y, h = g;
          for (let R = l - r; R !== l; R += 1)
            S(e[R][1]);
        } else
          for (let $ = d; $ !== l; $ += 1)
            S(e[$][1]);
      }
    for (; l !== n && e[l][0] === U; )
      _(e[l][1]), l += 1;
    for (; l !== n && e[l][0] === k; )
      O(e[l][1]), l += 1;
  }
  return m && (f[b] = Dn(y, u, g, h, t)), f.join(`
`);
}
c(Oi, "joinAlignedDiffsNoExpand");
function $i(e, t) {
  return e.map((n, r, o) => {
    let i = n[1], s = r === 0 || r === o.length - 1;
    switch (n[0]) {
      case U:
        return Jn(i, s, t);
      case k:
        return Xn(i, s, t);
      default:
        return Zn(i, s, t);
    }
  }).join(`
`);
}
c($i, "joinAlignedDiffsExpand");
var Tt = /* @__PURE__ */ c((e) => e, "noColor"), vn = 5, wi = 0;
function Ai() {
  return {
    aAnnotation: "Expected",
    aColor: ne.green,
    aIndicator: "-",
    bAnnotation: "Received",
    bColor: ne.red,
    bIndicator: "+",
    changeColor: ne.inverse,
    changeLineTrailingSpaceColor: Tt,
    commonColor: ne.dim,
    commonIndicator: " ",
    commonLineTrailingSpaceColor: Tt,
    compareKeys: void 0,
    contextLines: vn,
    emptyFirstOrLastLinePlaceholder: "",
    expand: !1,
    includeChangeCounts: !1,
    omitAnnotationLines: !1,
    patchColor: ne.yellow,
    printBasicPrototype: !1,
    truncateThreshold: wi,
    truncateAnnotation: "... Diff result is truncated",
    truncateAnnotationColor: Tt
  };
}
c(Ai, "getDefaultOptions");
function Ri(e) {
  return e && typeof e == "function" ? e : void 0;
}
c(Ri, "getCompareKeys");
function Ni(e) {
  return typeof e == "number" && Number.isSafeInteger(e) && e >= 0 ? e : vn;
}
c(Ni, "getContextLines");
function he(e = {}) {
  return {
    ...Ai(),
    ...e,
    compareKeys: Ri(e.compareKeys),
    contextLines: Ni(e.contextLines)
  };
}
c(he, "normalizeDiffOptions");
function Ee(e) {
  return e.length === 1 && e[0].length === 0;
}
c(Ee, "isEmptyString");
function Pi(e) {
  let t = 0, n = 0;
  return e.forEach((r) => {
    switch (r[0]) {
      case U:
        t += 1;
        break;
      case k:
        n += 1;
        break;
    }
  }), {
    a: t,
    b: n
  };
}
c(Pi, "countChanges");
function Ii({ aAnnotation: e, aColor: t, aIndicator: n, bAnnotation: r, bColor: o, bIndicator: i, includeChangeCounts: s, omitAnnotationLines: a }, l) {
  if (a)
    return "";
  let m = "", p = "";
  if (s) {
    let y = String(l.a), g = String(l.b), u = r.length - e.length, h = " ".repeat(Math.max(0, u)), S = " ".repeat(Math.max(0, -u)), _ = g.length -
    y.length, O = " ".repeat(Math.max(0, _)), d = " ".repeat(Math.max(0, -_));
    m = `${h}  ${n} ${O}${y}`, p = `${S}  ${i} ${d}${g}`;
  }
  let f = `${n} ${e}${m}`, b = `${i} ${r}${p}`;
  return `${t(f)}
${o(b)}

`;
}
c(Ii, "printAnnotation");
function Nt(e, t, n) {
  return Ii(n, Pi(e)) + (n.expand ? $i(e, n) : Oi(e, n)) + (t ? n.truncateAnnotationColor(`
${n.truncateAnnotation}`) : "");
}
c(Nt, "printDiffLines");
function Xe(e, t, n) {
  let r = he(n), [o, i] = Qn(Ee(e) ? [] : e, Ee(t) ? [] : t, r);
  return Nt(o, i, r);
}
c(Xe, "diffLinesUnified");
function Mi(e, t, n, r, o) {
  if (Ee(e) && Ee(n) && (e = [], n = []), Ee(t) && Ee(r) && (t = [], r = []), e.length !== n.length || t.length !== r.length)
    return Xe(e, t, o);
  let [i, s] = Qn(n, r, o), a = 0, l = 0;
  return i.forEach((m) => {
    switch (m[0]) {
      case U:
        m[1] = e[a], a += 1;
        break;
      case k:
        m[1] = t[l], l += 1;
        break;
      default:
        m[1] = t[l], a += 1, l += 1;
    }
  }), Nt(i, s, he(o));
}
c(Mi, "diffLinesUnified2");
function Qn(e, t, n) {
  let r = n?.truncateThreshold ?? !1, o = Math.max(Math.floor(n?.truncateThreshold ?? 0), 0), i = r ? Math.min(e.length, o) : e.length, s = r ?
  Math.min(t.length, o) : t.length, a = i !== e.length || s !== t.length, l = /* @__PURE__ */ c((y, g) => e[y] === t[g], "isCommon"), m = [],
  p = 0, f = 0;
  for (Hn(i, s, l, /* @__PURE__ */ c((y, g, u) => {
    for (; p !== g; p += 1)
      m.push(new I(U, e[p]));
    for (; f !== u; f += 1)
      m.push(new I(k, t[f]));
    for (; y !== 0; y -= 1, p += 1, f += 1)
      m.push(new I(x, t[f]));
  }, "foundSubsequence")); p !== i; p += 1)
    m.push(new I(U, e[p]));
  for (; f !== s; f += 1)
    m.push(new I(k, t[f]));
  return [m, a];
}
c(Qn, "diffLinesRaw");
function Fn(e) {
  if (e === void 0)
    return "undefined";
  if (e === null)
    return "null";
  if (Array.isArray(e))
    return "array";
  if (typeof e == "boolean")
    return "boolean";
  if (typeof e == "function")
    return "function";
  if (typeof e == "number")
    return "number";
  if (typeof e == "string")
    return "string";
  if (typeof e == "bigint")
    return "bigint";
  if (typeof e == "object") {
    if (e != null) {
      if (e.constructor === RegExp)
        return "regexp";
      if (e.constructor === Map)
        return "map";
      if (e.constructor === Set)
        return "set";
      if (e.constructor === Date)
        return "date";
    }
    return "object";
  } else if (typeof e == "symbol")
    return "symbol";
  throw new Error(`value of unknown type: ${e}`);
}
c(Fn, "getType");
function jn(e) {
  return e.includes(`\r
`) ? `\r
` : `
`;
}
c(jn, "getNewLineSymbol");
function Li(e, t, n) {
  let r = n?.truncateThreshold ?? !1, o = Math.max(Math.floor(n?.truncateThreshold ?? 0), 0), i = e.length, s = t.length;
  if (r) {
    let y = e.includes(`
`), g = t.includes(`
`), u = jn(e), h = jn(t), S = y ? `${e.split(u, o).join(u)}
` : e, _ = g ? `${t.split(h, o).join(h)}
` : t;
    i = S.length, s = _.length;
  }
  let a = i !== e.length || s !== t.length, l = /* @__PURE__ */ c((y, g) => e[y] === t[g], "isCommon"), m = 0, p = 0, f = [];
  return Hn(i, s, l, /* @__PURE__ */ c((y, g, u) => {
    m !== g && f.push(new I(U, e.slice(m, g))), p !== u && f.push(new I(k, t.slice(p, u))), m = g + y, p = u + y, f.push(new I(x, t.slice(u,
    p)));
  }, "foundSubsequence")), m !== i && f.push(new I(U, e.slice(m))), p !== s && f.push(new I(k, t.slice(p))), [f, a];
}
c(Li, "diffStrings");
function xi(e, t, n) {
  return t.reduce((r, o) => r + (o[0] === x ? o[1] : o[0] === e && o[1].length !== 0 ? n(o[1]) : ""), "");
}
c(xi, "concatenateRelevantDiffs");
var Je = class {
  static {
    c(this, "ChangeBuffer");
  }
  op;
  line;
  lines;
  changeColor;
  constructor(t, n) {
    this.op = t, this.line = [], this.lines = [], this.changeColor = n;
  }
  pushSubstring(t) {
    this.pushDiff(new I(this.op, t));
  }
  pushLine() {
    this.lines.push(this.line.length !== 1 ? new I(this.op, xi(this.op, this.line, this.changeColor)) : this.line[0][0] === this.op ? this.line[0] :
    new I(this.op, this.line[0][1])), this.line.length = 0;
  }
  isLineEmpty() {
    return this.line.length === 0;
  }
  pushDiff(t) {
    this.line.push(t);
  }
  align(t) {
    let n = t[1];
    if (n.includes(`
`)) {
      let r = n.split(`
`), o = r.length - 1;
      r.forEach((i, s) => {
        s < o ? (this.pushSubstring(i), this.pushLine()) : i.length !== 0 && this.pushSubstring(i);
      });
    } else
      this.pushDiff(t);
  }
  moveLinesTo(t) {
    this.isLineEmpty() || this.pushLine(), t.push(...this.lines), this.lines.length = 0;
  }
}, Ot = class {
  static {
    c(this, "CommonBuffer");
  }
  deleteBuffer;
  insertBuffer;
  lines;
  constructor(t, n) {
    this.deleteBuffer = t, this.insertBuffer = n, this.lines = [];
  }
  pushDiffCommonLine(t) {
    this.lines.push(t);
  }
  pushDiffChangeLines(t) {
    let n = t[1].length === 0;
    (!n || this.deleteBuffer.isLineEmpty()) && this.deleteBuffer.pushDiff(t), (!n || this.insertBuffer.isLineEmpty()) && this.insertBuffer.pushDiff(
    t);
  }
  flushChangeLines() {
    this.deleteBuffer.moveLinesTo(this.lines), this.insertBuffer.moveLinesTo(this.lines);
  }
  align(t) {
    let n = t[0], r = t[1];
    if (r.includes(`
`)) {
      let o = r.split(`
`), i = o.length - 1;
      o.forEach((s, a) => {
        if (a === 0) {
          let l = new I(n, s);
          this.deleteBuffer.isLineEmpty() && this.insertBuffer.isLineEmpty() ? (this.flushChangeLines(), this.pushDiffCommonLine(l)) : (this.
          pushDiffChangeLines(l), this.flushChangeLines());
        } else a < i ? this.pushDiffCommonLine(new I(n, s)) : s.length !== 0 && this.pushDiffChangeLines(new I(n, s));
      });
    } else
      this.pushDiffChangeLines(t);
  }
  getLines() {
    return this.flushChangeLines(), this.lines;
  }
};
function Di(e, t) {
  let n = new Je(U, t), r = new Je(k, t), o = new Ot(n, r);
  return e.forEach((i) => {
    switch (i[0]) {
      case U:
        n.align(i);
        break;
      case k:
        r.align(i);
        break;
      default:
        o.align(i);
    }
  }), o.getLines();
}
c(Di, "getAlignedDiffs");
function Fi(e, t) {
  if (t) {
    let n = e.length - 1;
    return e.some((r, o) => r[0] === x && (o !== n || r[1] !== `
`));
  }
  return e.some((n) => n[0] === x);
}
c(Fi, "hasCommonDiff");
function ji(e, t, n) {
  if (e !== t && e.length !== 0 && t.length !== 0) {
    let r = e.includes(`
`) || t.includes(`
`), [o, i] = er(r ? `${e}
` : e, r ? `${t}
` : t, !0, n);
    if (Fi(o, r)) {
      let s = he(n), a = Di(o, s.changeColor);
      return Nt(a, i, s);
    }
  }
  return Xe(e.split(`
`), t.split(`
`), n);
}
c(ji, "diffStringsUnified");
function er(e, t, n, r) {
  let [o, i] = Li(e, t, r);
  return n && yi(o), [o, i];
}
c(er, "diffStringsRaw");
function $t(e, t) {
  let { commonColor: n } = he(t);
  return n(e);
}
c($t, "getCommonMessage");
var { AsymmetricMatcher: Bi, DOMCollection: ki, DOMElement: zi, Immutable: Yi, ReactElement: Ui, ReactTestComponent: Wi } = $e, tr = [
  Wi,
  Ui,
  zi,
  ki,
  Yi,
  Bi,
  $e.Error
], wt = {
  maxDepth: 20,
  plugins: tr
}, nr = {
  callToJSON: !1,
  maxDepth: 8,
  plugins: tr
};
function Vi(e, t, n) {
  if (Object.is(e, t))
    return "";
  let r = Fn(e), o = r, i = !1;
  if (r === "object" && typeof e.asymmetricMatch == "function") {
    if (e.$$typeof !== Symbol.for("jest.asymmetricMatcher") || typeof e.getExpectedType != "function")
      return;
    o = e.getExpectedType(), i = o === "string";
  }
  if (o !== Fn(t)) {
    let h = function(O) {
      return O.length <= u ? O : `${O.slice(0, u)}...`;
    };
    c(h, "truncate");
    let { aAnnotation: s, aColor: a, aIndicator: l, bAnnotation: m, bColor: p, bIndicator: f } = he(n), b = At(nr, n), y = Q(e, b), g = Q(t,
    b), u = 1e5;
    y = h(y), g = h(g);
    let S = `${a(`${l} ${s}:`)} 
${y}`, _ = `${p(`${f} ${m}:`)} 
${g}`;
    return `${S}

${_}`;
  }
  if (!i)
    switch (r) {
      case "string":
        return Xe(e.split(`
`), t.split(`
`), n);
      case "boolean":
      case "number":
        return qi(e, t, n);
      case "map":
        return Ct(Bn(e), Bn(t), n);
      case "set":
        return Ct(kn(e), kn(t), n);
      default:
        return Ct(e, t, n);
    }
}
c(Vi, "diff");
function qi(e, t, n) {
  let r = Q(e, wt), o = Q(t, wt);
  return r === o ? "" : Xe(r.split(`
`), o.split(`
`), n);
}
c(qi, "comparePrimitive");
function Bn(e) {
  return new Map(Array.from(e.entries()).sort());
}
c(Bn, "sortMap");
function kn(e) {
  return new Set(Array.from(e.values()).sort());
}
c(kn, "sortSet");
function Ct(e, t, n) {
  let r, o = !1;
  try {
    let s = At(wt, n);
    r = zn(e, t, s, n);
  } catch {
    o = !0;
  }
  let i = $t(Gn, n);
  if (r === void 0 || r === i) {
    let s = At(nr, n);
    r = zn(e, t, s, n), r !== i && !o && (r = `${$t(Ei, n)}

${r}`);
  }
  return r;
}
c(Ct, "compareObjects");
function At(e, t) {
  let { compareKeys: n, printBasicPrototype: r, maxDepth: o } = he(t);
  return {
    ...e,
    compareKeys: n,
    printBasicPrototype: r,
    maxDepth: o ?? e.maxDepth
  };
}
c(At, "getFormatOptions");
function zn(e, t, n, r) {
  let o = {
    ...n,
    indent: 0
  }, i = Q(e, o), s = Q(t, o);
  if (i === s)
    return $t(Gn, r);
  {
    let a = Q(e, n), l = Q(t, n);
    return Mi(a.split(`
`), l.split(`
`), i.split(`
`), s.split(`
`), r);
  }
}
c(zn, "getObjectsDifference");
var Yn = 2e4;
function Un(e) {
  return Ke(e) === "Object" && typeof e.asymmetricMatch == "function";
}
c(Un, "isAsymmetricMatcher");
function Wn(e, t) {
  let n = Ke(e), r = Ke(t);
  return n === r && (n === "Object" || n === "Array");
}
c(Wn, "isReplaceable");
function rr(e, t, n) {
  let { aAnnotation: r, bAnnotation: o } = he(n);
  if (typeof t == "string" && typeof e == "string" && t.length > 0 && e.length > 0 && t.length <= Yn && e.length <= Yn && t !== e) {
    if (t.includes(`
`) || e.includes(`
`))
      return ji(t, e, n);
    let [p] = er(t, e, !0), f = p.some((u) => u[0] === x), b = Ki(r, o), y = b(r) + Ji(Vn(p, U, f)), g = b(o) + Hi(Vn(p, k, f));
    return `${y}
${g}`;
  }
  let i = _t(t, { forceWritable: !0 }), s = _t(e, { forceWritable: !0 }), { replacedExpected: a, replacedActual: l } = or(s, i);
  return Vi(a, l, n);
}
c(rr, "printDiffOrStringify");
function or(e, t, n = /* @__PURE__ */ new WeakSet(), r = /* @__PURE__ */ new WeakSet()) {
  return e instanceof Error && t instanceof Error && typeof e.cause < "u" && typeof t.cause > "u" ? (delete e.cause, {
    replacedActual: e,
    replacedExpected: t
  }) : Wn(e, t) ? n.has(e) || r.has(t) ? {
    replacedActual: e,
    replacedExpected: t
  } : (n.add(e), r.add(t), Et(t).forEach((o) => {
    let i = t[o], s = e[o];
    if (Un(i))
      i.asymmetricMatch(s) && (e[o] = i);
    else if (Un(s))
      s.asymmetricMatch(i) && (t[o] = s);
    else if (Wn(s, i)) {
      let a = or(s, i, n, r);
      e[o] = a.replacedActual, t[o] = a.replacedExpected;
    }
  }), {
    replacedActual: e,
    replacedExpected: t
  }) : {
    replacedActual: e,
    replacedExpected: t
  };
}
c(or, "replaceAsymmetricMatcher");
function Ki(...e) {
  let t = e.reduce((n, r) => r.length > n ? r.length : n, 0);
  return (n) => `${n}: ${" ".repeat(t - n.length)}`;
}
c(Ki, "getLabelPrinter");
var Gi = "\xB7";
function ir(e) {
  return e.replace(/\s+$/gm, (t) => Gi.repeat(t.length));
}
c(ir, "replaceTrailingSpaces");
function Hi(e) {
  return ne.red(ir(ge(e)));
}
c(Hi, "printReceived");
function Ji(e) {
  return ne.green(ir(ge(e)));
}
c(Ji, "printExpected");
function Vn(e, t, n) {
  return e.reduce((r, o) => r + (o[0] === x ? o[1] : o[0] === t ? n ? ne.inverse(o[1]) : o[1] : ""), "");
}
c(Vn, "getCommonAndChangedSubstrings");

// ../node_modules/@vitest/utils/dist/error.js
var Xi = "@@__IMMUTABLE_RECORD__@@", Zi = "@@__IMMUTABLE_ITERABLE__@@";
function vi(e) {
  return e && (e[Zi] || e[Xi]);
}
c(vi, "isImmutable");
var Qi = Object.getPrototypeOf({});
function sr(e) {
  return e instanceof Error ? `<unserializable>: ${e.message}` : typeof e == "string" ? `<unserializable>: ${e}` : "<unserializable>";
}
c(sr, "getUnserializableMessage");
function _e(e, t = /* @__PURE__ */ new WeakMap()) {
  if (!e || typeof e == "string")
    return e;
  if (typeof e == "function")
    return `Function<${e.name || "anonymous"}>`;
  if (typeof e == "symbol")
    return e.toString();
  if (typeof e != "object")
    return e;
  if (vi(e))
    return _e(e.toJSON(), t);
  if (e instanceof Promise || e.constructor && e.constructor.prototype === "AsyncFunction")
    return "Promise";
  if (typeof Element < "u" && e instanceof Element)
    return e.tagName;
  if (typeof e.asymmetricMatch == "function")
    return `${e.toString()} ${An(e.sample)}`;
  if (typeof e.toJSON == "function")
    return _e(e.toJSON(), t);
  if (t.has(e))
    return t.get(e);
  if (Array.isArray(e)) {
    let n = new Array(e.length);
    return t.set(e, n), e.forEach((r, o) => {
      try {
        n[o] = _e(r, t);
      } catch (i) {
        n[o] = sr(i);
      }
    }), n;
  } else {
    let n = /* @__PURE__ */ Object.create(null);
    t.set(e, n);
    let r = e;
    for (; r && r !== Qi; )
      Object.getOwnPropertyNames(r).forEach((o) => {
        if (!(o in n))
          try {
            n[o] = _e(e[o], t);
          } catch (i) {
            delete n[o], n[o] = sr(i);
          }
      }), r = Object.getPrototypeOf(r);
    return n;
  }
}
c(_e, "serializeValue");
function es(e) {
  return e.replace(/__(vite_ssr_import|vi_import)_\d+__\./g, "");
}
c(es, "normalizeErrorMessage");
function Pt(e, t, n = /* @__PURE__ */ new WeakSet()) {
  if (!e || typeof e != "object")
    return { message: String(e) };
  let r = e;
  r.stack && (r.stackStr = String(r.stack)), r.name && (r.nameStr = String(r.name)), (r.showDiff || r.showDiff === void 0 && r.expected !== void 0 &&
  r.actual !== void 0) && (r.diff = rr(r.actual, r.expected, {
    ...t,
    ...r.diffOptions
  })), typeof r.expected != "string" && (r.expected = ge(r.expected, 10)), typeof r.actual != "string" && (r.actual = ge(r.actual, 10));
  try {
    typeof r.message == "string" && (r.message = es(r.message));
  } catch {
  }
  try {
    !n.has(r) && typeof r.cause == "object" && (n.add(r), r.cause = Pt(r.cause, t, n));
  } catch {
  }
  try {
    return _e(r);
  } catch (o) {
    return _e(new Error(`Failed to fully serialize error: ${o?.message}
Inner error message: ${r?.message}`));
  }
}
c(Pt, "processError");

// src/instrumenter/EVENTS.ts
var ee = {
  CALL: "storybook/instrumenter/call",
  SYNC: "storybook/instrumenter/sync",
  START: "storybook/instrumenter/start",
  BACK: "storybook/instrumenter/back",
  GOTO: "storybook/instrumenter/goto",
  NEXT: "storybook/instrumenter/next",
  END: "storybook/instrumenter/end"
};

// src/instrumenter/preview-api.ts
var Ze = globalThis.__STORYBOOK_ADDONS_PREVIEW;

// src/instrumenter/types.ts
var cr = /* @__PURE__ */ ((o) => (o.DONE = "done", o.ERROR = "error", o.ACTIVE = "active", o.WAITING = "waiting", o))(cr || {});

// src/instrumenter/instrumenter.ts
var ur = {
  start: !1,
  back: !1,
  goto: !1,
  next: !1,
  end: !1
}, ts = new Error(
  "This function ran after the play function completed. Did you forget to `await` it?"
), lr = /* @__PURE__ */ c((e) => Object.prototype.toString.call(e) === "[object Object]", "isObject"), ns = /* @__PURE__ */ c((e) => Object.
prototype.toString.call(e) === "[object Module]", "isModule"), rs = /* @__PURE__ */ c((e) => {
  if (!lr(e) && !ns(e))
    return !1;
  if (e.constructor === void 0)
    return !0;
  let t = e.constructor.prototype;
  return !!lr(t);
}, "isInstrumentable"), os = /* @__PURE__ */ c((e) => {
  try {
    return new e.constructor();
  } catch {
    return {};
  }
}, "construct"), It = /* @__PURE__ */ c(() => ({
  renderPhase: void 0,
  isDebugging: !1,
  isPlaying: !1,
  isLocked: !1,
  cursor: 0,
  calls: [],
  shadowCalls: [],
  callRefsByResult: /* @__PURE__ */ new Map(),
  chainedCallIds: /* @__PURE__ */ new Set(),
  ancestors: [],
  playUntil: void 0,
  resolvers: {},
  syncTimeout: void 0
}), "getInitialState"), ar = /* @__PURE__ */ c((e, t = !1) => {
  let n = (t ? e.shadowCalls : e.calls).filter((o) => o.retain);
  if (!n.length)
    return;
  let r = new Map(
    Array.from(e.callRefsByResult.entries()).filter(([, o]) => o.retain)
  );
  return { cursor: n.length, calls: n, callRefsByResult: r };
}, "getRetainedState"), Mt = class {
  constructor() {
    this.initialized = !1;
    this.state = W.global.window?.parent?.__STORYBOOK_ADDON_INTERACTIONS_INSTRUMENTER_STATE__ || {};
    let t = /* @__PURE__ */ c(({
      storyId: l,
      isPlaying: m = !0,
      isDebugging: p = !1
    }) => {
      let f = this.getState(l);
      this.setState(l, {
        ...It(),
        ...ar(f, p),
        shadowCalls: p ? f.shadowCalls : [],
        chainedCallIds: p ? f.chainedCallIds : /* @__PURE__ */ new Set(),
        playUntil: p ? f.playUntil : void 0,
        isPlaying: m,
        isDebugging: p
      }), this.sync(l);
    }, "resetState"), n = /* @__PURE__ */ c((l) => ({ storyId: m, playUntil: p }) => {
      this.getState(m).isDebugging || this.setState(m, ({ calls: b }) => ({
        calls: [],
        shadowCalls: b.map((y) => ({ ...y, status: "waiting" })),
        isDebugging: !0
      }));
      let f = this.getLog(m);
      this.setState(m, ({ shadowCalls: b }) => {
        if (p || !f.length)
          return { playUntil: p };
        let y = b.findIndex((g) => g.id === f[0].callId);
        return {
          playUntil: b.slice(0, y).filter((g) => g.interceptable && !g.ancestors?.length).slice(-1)[0]?.id
        };
      }), l.emit(ye.FORCE_REMOUNT, { storyId: m, isDebugging: !0 });
    }, "start"), r = /* @__PURE__ */ c((l) => ({ storyId: m }) => {
      let p = this.getLog(m).filter((b) => !b.ancestors?.length), f = p.reduceRight((b, y, g) => b >= 0 || y.status === "waiting" ? b : g, -1);
      n(l)({ storyId: m, playUntil: p[f - 1]?.callId });
    }, "back"), o = /* @__PURE__ */ c((l) => ({ storyId: m, callId: p }) => {
      let { calls: f, shadowCalls: b, resolvers: y } = this.getState(m), g = f.find(({ id: h }) => h === p), u = b.find(({ id: h }) => h ===
      p);
      if (!g && u && Object.values(y).length > 0) {
        let h = this.getLog(m).find((S) => S.status === "waiting")?.callId;
        u.id !== h && this.setState(m, { playUntil: u.id }), Object.values(y).forEach((S) => S());
      } else
        n(l)({ storyId: m, playUntil: p });
    }, "goto"), i = /* @__PURE__ */ c((l) => ({ storyId: m }) => {
      let { resolvers: p } = this.getState(m);
      if (Object.values(p).length > 0)
        Object.values(p).forEach((f) => f());
      else {
        let f = this.getLog(m).find((b) => b.status === "waiting")?.callId;
        f ? n(l)({ storyId: m, playUntil: f }) : s({ storyId: m });
      }
    }, "next"), s = /* @__PURE__ */ c(({ storyId: l }) => {
      this.setState(l, { playUntil: void 0, isDebugging: !1 }), Object.values(this.getState(l).resolvers).forEach((m) => m());
    }, "end"), a = /* @__PURE__ */ c(({ storyId: l, newPhase: m }) => {
      let { isDebugging: p } = this.getState(l);
      this.setState(l, { renderPhase: m }), m === "preparing" && p && t({ storyId: l }), m === "playing" && t({ storyId: l, isDebugging: p }),
      m === "played" && this.setState(l, {
        isLocked: !1,
        isPlaying: !1,
        isDebugging: !1
      }), m === "errored" && this.setState(l, {
        isLocked: !1,
        isPlaying: !1
      });
    }, "renderPhaseChanged");
    Ze && Ze.ready().then(() => {
      this.channel = Ze.getChannel(), this.channel.on(ye.FORCE_REMOUNT, t), this.channel.on(ye.STORY_RENDER_PHASE_CHANGED, a), this.channel.
      on(ye.SET_CURRENT_STORY, () => {
        this.initialized ? this.cleanup() : this.initialized = !0;
      }), this.channel.on(ee.START, n(this.channel)), this.channel.on(ee.BACK, r(this.channel)), this.channel.on(ee.GOTO, o(this.channel)), this.
      channel.on(ee.NEXT, i(this.channel)), this.channel.on(ee.END, s);
    });
  }
  static {
    c(this, "Instrumenter");
  }
  getState(t) {
    return this.state[t] || It();
  }
  setState(t, n) {
    let r = this.getState(t), o = typeof n == "function" ? n(r) : n;
    this.state = { ...this.state, [t]: { ...r, ...o } }, W.global.window?.parent && (W.global.window.parent.__STORYBOOK_ADDON_INTERACTIONS_INSTRUMENTER_STATE__ =
    this.state);
  }
  cleanup() {
    this.state = Object.entries(this.state).reduce(
      (n, [r, o]) => {
        let i = ar(o);
        return i && (n[r] = Object.assign(It(), i)), n;
      },
      {}
    );
    let t = { controlStates: ur, logItems: [] };
    this.channel?.emit(ee.SYNC, t), W.global.window?.parent && (W.global.window.parent.__STORYBOOK_ADDON_INTERACTIONS_INSTRUMENTER_STATE__ =
    this.state);
  }
  getLog(t) {
    let { calls: n, shadowCalls: r } = this.getState(t), o = [...r];
    n.forEach((s, a) => {
      o[a] = s;
    });
    let i = /* @__PURE__ */ new Set();
    return o.reduceRight((s, a) => (a.args.forEach((l) => {
      l?.__callId__ && i.add(l.__callId__);
    }), a.path.forEach((l) => {
      l.__callId__ && i.add(l.__callId__);
    }), (a.interceptable || a.exception) && !i.has(a.id) && (s.unshift({ callId: a.id, status: a.status, ancestors: a.ancestors }), i.add(a.
    id)), s), []);
  }
  // Traverses the object structure to recursively patch all function properties.
  // Returns the original object, or a new object with the same constructor,
  // depending on whether it should mutate.
  instrument(t, n, r = 0) {
    if (!rs(t))
      return t;
    let { mutate: o = !1, path: i = [] } = n, s = n.getKeys ? n.getKeys(t, r) : Object.keys(t);
    return r += 1, s.reduce(
      (a, l) => {
        let m = is(t, l);
        if (typeof m?.get == "function") {
          if (m.configurable) {
            let f = /* @__PURE__ */ c(() => m?.get?.bind(t)?.(), "getter");
            Object.defineProperty(a, l, {
              get: /* @__PURE__ */ c(() => this.instrument(f(), { ...n, path: i.concat(l) }, r), "get")
            });
          }
          return a;
        }
        let p = t[l];
        return typeof p != "function" ? (a[l] = this.instrument(p, { ...n, path: i.concat(l) }, r), a) : "__originalFn__" in p && typeof p.__originalFn__ ==
        "function" ? (a[l] = p, a) : (a[l] = (...f) => this.track(l, p, t, f, n), a[l].__originalFn__ = p, Object.defineProperty(a[l], "name",
        { value: l, writable: !1 }), Object.keys(p).length > 0 && Object.assign(
          a[l],
          this.instrument({ ...p }, { ...n, path: i.concat(l) }, r)
        ), a);
      },
      o ? t : os(t)
    );
  }
  // Monkey patch an object method to record calls.
  // Returns a function that invokes the original function, records the invocation ("call") and
  // returns the original result.
  track(t, n, r, o, i) {
    let s = o?.[0]?.__storyId__ || W.global.__STORYBOOK_PREVIEW__?.selectionStore?.selection?.storyId, { cursor: a, ancestors: l } = this.getState(
    s);
    this.setState(s, { cursor: a + 1 });
    let m = `${l.slice(-1)[0] || s} [${a}] ${t}`, { path: p = [], intercept: f = !1, retain: b = !1 } = i, y = typeof f == "function" ? f(t,
    p) : f, g = { id: m, cursor: a, storyId: s, ancestors: l, path: p, method: t, args: o, interceptable: y, retain: b }, h = (y && !l.length ?
    this.intercept : this.invoke).call(this, n, r, g, i);
    return this.instrument(h, { ...i, mutate: !0, path: [{ __callId__: g.id }] });
  }
  intercept(t, n, r, o) {
    let { chainedCallIds: i, isDebugging: s, playUntil: a } = this.getState(r.storyId), l = i.has(r.id);
    return !s || l || a ? (a === r.id && this.setState(r.storyId, { playUntil: void 0 }), this.invoke(t, n, r, o)) : new Promise((m) => {
      this.setState(r.storyId, ({ resolvers: p }) => ({
        isLocked: !1,
        resolvers: { ...p, [r.id]: m }
      }));
    }).then(() => (this.setState(r.storyId, (m) => {
      let { [r.id]: p, ...f } = m.resolvers;
      return { isLocked: !0, resolvers: f };
    }), this.invoke(t, n, r, o)));
  }
  invoke(t, n, r, o) {
    let { callRefsByResult: i, renderPhase: s } = this.getState(r.storyId), a = 25, l = /* @__PURE__ */ c((f, b, y) => {
      if (y.includes(f))
        return "[Circular]";
      if (y = [...y, f], b > a)
        return "...";
      if (i.has(f))
        return i.get(f);
      if (f instanceof Array)
        return f.map((g) => l(g, ++b, y));
      if (f instanceof Date)
        return { __date__: { value: f.toISOString() } };
      if (f instanceof Error) {
        let { name: g, message: u, stack: h } = f;
        return { __error__: { name: g, message: u, stack: h } };
      }
      if (f instanceof RegExp) {
        let { flags: g, source: u } = f;
        return { __regexp__: { flags: g, source: u } };
      }
      if (f instanceof W.global.window?.HTMLElement) {
        let { prefix: g, localName: u, id: h, classList: S, innerText: _ } = f, O = Array.from(S);
        return { __element__: { prefix: g, localName: u, id: h, classNames: O, innerText: _ } };
      }
      return typeof f == "function" ? {
        __function__: { name: "getMockName" in f ? f.getMockName() : f.name }
      } : typeof f == "symbol" ? { __symbol__: { description: f.description } } : typeof f == "object" && f?.constructor?.name && f?.constructor?.
      name !== "Object" ? { __class__: { name: f.constructor.name } } : Object.prototype.toString.call(f) === "[object Object]" ? Object.fromEntries(
        Object.entries(f).map(([g, u]) => [g, l(u, ++b, y)])
      ) : f;
    }, "serializeValues"), m = {
      ...r,
      args: r.args.map((f) => l(f, 0, []))
    };
    r.path.forEach((f) => {
      f?.__callId__ && this.setState(r.storyId, ({ chainedCallIds: b }) => ({
        chainedCallIds: new Set(Array.from(b).concat(f.__callId__))
      }));
    });
    let p = /* @__PURE__ */ c((f) => {
      if (f instanceof Error) {
        let { name: b, message: y, stack: g, callId: u = r.id } = f, {
          showDiff: h = void 0,
          diff: S = void 0,
          actual: _ = void 0,
          expected: O = void 0
        } = f.name === "AssertionError" ? Pt(f) : f, d = { name: b, message: y, stack: g, callId: u, showDiff: h, diff: S, actual: _, expected: O };
        if (this.update({ ...m, status: "error", exception: d }), this.setState(r.storyId, (E) => ({
          callRefsByResult: new Map([
            ...Array.from(E.callRefsByResult.entries()),
            [f, { __callId__: r.id, retain: r.retain }]
          ])
        })), r.ancestors?.length)
          throw Object.prototype.hasOwnProperty.call(f, "callId") || Object.defineProperty(f, "callId", { value: r.id }), f;
      }
      throw f;
    }, "handleException");
    try {
      if (s === "played" && !r.retain)
        throw ts;
      let b = (o.getArgs ? o.getArgs(r, this.getState(r.storyId)) : r.args).map((g) => typeof g != "function" || ss(g) || Object.keys(g).length ?
      g : (...u) => {
        let { cursor: h, ancestors: S } = this.getState(r.storyId);
        this.setState(r.storyId, { cursor: 0, ancestors: [...S, r.id] });
        let _ = /* @__PURE__ */ c(() => this.setState(r.storyId, { cursor: h, ancestors: S }), "restore"), O = !1;
        try {
          let d = g(...u);
          return d instanceof Promise ? (O = !0, d.finally(_)) : d;
        } finally {
          O || _();
        }
      }), y = t.apply(n, b);
      return y && ["object", "function", "symbol"].includes(typeof y) && this.setState(r.storyId, (g) => ({
        callRefsByResult: new Map([
          ...Array.from(g.callRefsByResult.entries()),
          [y, { __callId__: r.id, retain: r.retain }]
        ])
      })), this.update({
        ...m,
        status: y instanceof Promise ? "active" : "done"
      }), y instanceof Promise ? y.then((g) => (this.update({ ...m, status: "done" }), g), p) : y;
    } catch (f) {
      return p(f);
    }
  }
  // Sends the call info to the manager and synchronizes the log.
  update(t) {
    this.channel?.emit(ee.CALL, t), this.setState(t.storyId, ({ calls: n }) => {
      let r = n.concat(t).reduce((o, i) => Object.assign(o, { [i.id]: i }), {});
      return {
        // Calls are sorted to ensure parent calls always come before calls in their callback.
        calls: Object.values(r).sort(
          (o, i) => o.id.localeCompare(i.id, void 0, { numeric: !0 })
        )
      };
    }), this.sync(t.storyId);
  }
  // Builds a log of interceptable calls and control states and sends it to the manager.
  // Uses a 0ms debounce because this might get called many times in one tick.
  sync(t) {
    let n = /* @__PURE__ */ c(() => {
      let { isLocked: r, isPlaying: o } = this.getState(t), i = this.getLog(t), s = i.filter(({ ancestors: f }) => !f.length).find((f) => f.
      status === "waiting")?.callId, a = i.some((f) => f.status === "active");
      if (r || a || i.length === 0) {
        let f = { controlStates: ur, logItems: i };
        this.channel?.emit(ee.SYNC, f);
        return;
      }
      let l = i.some(
        (f) => f.status === "done" || f.status === "error"
      ), p = { controlStates: {
        start: l,
        back: l,
        goto: !0,
        next: o,
        end: o
      }, logItems: i, pausedAt: s };
      this.channel?.emit(ee.SYNC, p);
    }, "synchronize");
    this.setState(t, ({ syncTimeout: r }) => (clearTimeout(r), { syncTimeout: setTimeout(n, 0) }));
  }
};
function mr(e, t = {}) {
  try {
    let n = !1, r = !1;
    return W.global.window?.location?.search?.includes("instrument=true") ? n = !0 : W.global.window?.location?.search?.includes("instrument\
=false") && (r = !0), W.global.window?.parent === W.global.window && !n || r ? e : (W.global.window && !W.global.window.__STORYBOOK_ADDON_INTERACTIONS_INSTRUMENTER__ &&
    (W.global.window.__STORYBOOK_ADDON_INTERACTIONS_INSTRUMENTER__ = new Mt()), (W.global.window?.__STORYBOOK_ADDON_INTERACTIONS_INSTRUMENTER__).
    instrument(e, t));
  } catch (n) {
    return fr.once.warn(n), e;
  }
}
c(mr, "instrument");
function is(e, t) {
  let n = e;
  for (; n != null; ) {
    let r = Object.getOwnPropertyDescriptor(n, t);
    if (r)
      return r;
    n = Object.getPrototypeOf(n);
  }
}
c(is, "getPropertyDescriptor");
function ss(e) {
  if (typeof e != "function")
    return !1;
  let t = Object.getOwnPropertyDescriptor(e, "prototype");
  return t ? !t.writable : !1;
}
c(ss, "isClass");
